---
title: "Figures"
author: "Chau"
date: "2024-04-29"
output: html_document
editor_options: 
  chunk_output_type: inline
---


```{r, include= FALSE, message=FALSE, error=FALSE, warning=FALSE}
library(Seurat)
library(ggplot2)
library(plotly)
library(cowplot)
library(dplyr)
library(tidyr)
library(ggrepel)
library(stringr)
library(DT)
library(paletteer)
library(viridisLite)
library(viridis)
library(scigenex)
scigenex::set_verbosity(0)
library(tibble)
library(circlize)
library(tidyverse)
library(AUCell)
library(clusterProfiler)
library(msigdbr)
library(fgsea)
library(enrichplot)
library(DOSE)
library(gridExtra)
library(patchwork)
library(data.table)
library(ggthemes)
library(ggalluvial)
library(org.Mm.eg.db)
library(circlize)
library(fmsb)
library(VennDiagram)
library(reshape2)
library(pheatmap)
if (!requireNamespace("fmsb", quietly = TRUE)) {
  install.packages("fmsb")
}
#BiocManager::install("org.Mm.eg.db")

if (!requireNamespace("VennDiagram", quietly = TRUE)) {
    install.packages("VennDiagram")
}

if (!requireNamespace("msigdbr", quietly = TRUE)) {
    install.packages("msigdbr")
}


source("/mnt/NASBIOINFO_CP/LALT/BIOINFO/PTEN/fonctions_R/DEG_functions.R")
```

```{r}
load(file = "/mnt/NASBIOINFO_CP/LALT/BIOINFO/PTEN/03_Analysis/Analysis2024/RObject/B6/merge_B6ccregress_2024.Robj")
load(file = "/mnt/NASBIOINFO_CP/LALT/BIOINFO/PTEN/03_Analysis/Analysis2024/RObject/BC/merge_2023exp_reg_cc.Robj")
load(file = "/mnt/NASBIOINFO_CP/LALT/BIOINFO/PTEN/03_Analysis/Analysis2024/RObject/B6/miceWT_B6_merged.Robj")
B6_WT <- miceWT
rm(miceWT)
load(file = "/mnt/NASBIOINFO_CP/LALT/BIOINFO/PTEN/03_Analysis/Analysis2024/RObject/BC/miceWT_2023_merged.Robj")
BC_WT <- miceWT
rm(miceWT)
```

# UMAP colored by tissu
```{r}
DimPlot(b6.combined.cc, group.by = "orig.ident", pt.size = 0.2, cols = c("#FF7676FF","#75B7D1FF"))
DimPlot(bc.combined.cc, group.by = "orig.ident", pt.size = 0.2, cols = c("#FF7676FF","#75B7D1FF","#7CAB7DFF"), reduction = "wnn.umap")

# UMAP of B6 ADT
DimPlot(b6.combined.cc, group.by = "orig.ident", pt.size = 0.2, cols = c("#FF7676FF","#75B7D1FF"), reduction = "adt.umap")
# UMAP of B6 RNA
DimPlot(bc.combined.cc, group.by = "orig.ident", pt.size = 0.2, cols = c("#FF7676FF","#75B7D1FF","#7CAB7DFF"), reduction = "rna.umap")
DimPlot(bc.combined.cc, group.by = "orig.ident", pt.size = 0.2, cols = c("#FF7676FF","#75B7D1FF","#7CAB7DFF"), reduction = "adt.umap")
```

```{r}
VlnPlot(bc.combined.cc,features = "nFeature_RNA", group.by = "orig.ident", pt.size = 0.1, cols = c("#FF7676FF","#75B7D1FF","#7CAB7DFF"))
VlnPlot(b6.combined.cc,features = "nFeature_RNA", group.by = "orig.ident", pt.size = 0.1, cols = c("#FF7676FF","#75B7D1FF"))
```

# UMAP colored by cell phase
```{r fig.width=15, fig.height=8, warning=FALSE, message=FALSE}
DimPlot(b6.combined.cc, group.by = "Phase", pt.size = 0.2) + DimPlot(bc.combined.cc, group.by = "Phase", pt.size = 0.2, reduction="wnn.umap")
```
# UMAP colored by leukestage
```{r fig.width=15, fig.height=8, warning=FALSE, message=FALSE}
bc.combined.cc$annotation <- factor(bc.combined.cc$annotation, levels=c("Thymus-WT","Thymus-PreTum early","Thymus-PreTum late","Thymus-Tum","Spleen-WT","Spleen-PreTum early","Spleen-PreTum late","Spleen-Tum"))
#leukestage.palette1 <- c("#9AC4F8","#7EB488","#fac341","#BB0A21","#9AC4F8","#7EB488","#fac341","#BB0A21")
leukestage.palette1 <- c("#9AC4F8","#fac341","#fac341","#BB0A21","#9AC4F8","#fac341","#fac341","#BB0A21") #classer PreTum early = PreTum late
names(leukestage.palette1) <- c("Thymus-WT","Thymus-PreTum early","Thymus-PreTum late","Thymus-Tum","Spleen-WT","Spleen-PreTum early","Spleen-PreTum late","Spleen-Tum")



b6.combined.cc$stage <- factor(b6.combined.cc$stage, levels=c("Thymus-WT","Thymus-PreTum","Thymus-Tum","Spleen-WT","Spleen-PreTum","Spleen-Tum"))
leukestage.palette2 <- c("#9AC4F8","#fac341","#BB0A21","#9AC4F8","#fac341","#BB0A21")
names(leukestage.palette2) <- c("Thymus-WT","Thymus-PreTum","Thymus-Tum","Spleen-WT","Spleen-PreTum","Spleen-Tum")
DimPlot(b6.combined.cc, group.by = "stage", pt.size = 0.2, cols = leukestage.palette2 ) + DimPlot(bc.combined.cc, group.by = "annotation", pt.size = 0.2,reduction = "wnn.umap", cols = leukestage.palette1 )

```

```{r}
bc.combined.cc@meta.data$annotation[which(bc.combined.cc@meta.data$annotation =="Thymus-PreTum early")] <- "Thymus-PreTum late"
bc.combined.cc@meta.data$annotation[which(bc.combined.cc@meta.data$annotation %in% "Spleen-PreTum early")] <- "Spleen-PreTum late"
```

# UMAP colored by tissu
```{r fig.width=15, fig.height=8, warning=FALSE, message=FALSE}
ogran.palette1 <- c("#8785B2FF","#8785B2FF","#8785B2FF","#8785B2FF","#D95F30FF","#D95F30FF","#D95F30FF","#D95F30FF")
names(ogran.palette1) <- c("Thymus-WT","Thymus-PreTum early","Thymus-PreTum late","Thymus-Tum","Spleen-WT","Spleen-PreTum early","Spleen-PreTum late","Spleen-Tum")
organ.palette2 <- c("#8785B2FF","#8785B2FF","#8785B2FF","#D95F30FF","#D95F30FF","#D95F30FF")
names(organ.palette2) <- c("Thymus-WT","Thymus-PreTum","Thymus-Tum","Spleen-WT","Spleen-PreTum","Spleen-Tum")
DimPlot(b6.combined.cc, group.by = "stage", pt.size = 0.2,reduction = "umap", cols = organ.palette2 )+ DimPlot(bc.combined.cc, group.by = "annotation", pt.size = 0.2,reduction = "wnn.umap", cols = ogran.palette1 )
```

# Dotplot : expression of marker genes and surface proteins in clusters of BALB/c Wild Type
```{r fig.width=8, fig.height=12, warning=FALSE, message=FALSE}
Idents(BC_WT)<-"wsnn_res.0.6"
BC_WT@active.ident <- factor(BC_WT@active.ident,levels=c("6","11","14","5","0","7","2","1","15","10","4","9","3","8","12","13"))
genes <- c("Il2ra","adt_CD25","Ptcra","Top2a","Mki67","Cdk1","Rag1","Rag2","Cd4","adt_CD4","Cd8a","adt_CD8","Cd27","Cd69","adt_CD69","Cd5","Tox2","Ccr7","adt_CCR7","S1pr1","Sell","Cxcr3","Ccl5","Gzma","Nkg7","Klra1","Trdc","adt_TCRgd","Foxp3","Trp53inp1","Bmf","percent.mt")
dp <- DotPlot(BC_WT, features = genes) + geom_point(aes(size=pct.exp), shape = 21, colour="black", stroke=0.5) + scale_colour_gradient2(low = "steelblue", mid = "ivory1", high = "red") + guides(size=guide_legend(override.aes=list(shape=21, colour="black", fill="white"))) + RotatedAxis()+coord_flip() 
dp + labs(title = "Expression of marker genes and surface proteins in clusters of BALB/c Wild Type")
```
# BC (WT) subset
```{r}
DimPlot(BC_WT, group.by = "wsnn_res.0.6",reduction = "wnn.umap" , pt.size = 0.5, cols = paletteer::paletteer_d("ggthemes::Hue_Circle"), label = TRUE)
```
# BC (WT) cluster annotation
```{r}
WT.palette <- rev( paletteer::paletteer_d("RColorBrewer::Spectral", n=11)  ) 
names(WT.palette) <- c("DN", "DPsm-CD25+", "DPsm-CCR7+", "DPblast", "DPsm", "DP69int", "DP69+", "SP4","SP8", "Tgd", "NK")
BC_WT$manual.annotation <- factor(BC_WT$manual.annotation, levels = c("DN", "DPsm-CD25+", "DPsm-CCR7+", "DPblast", "DPsm", "DP69int", "DP69+", "SP4","SP8", "Tgd", "NK"))
DimPlot(BC_WT,reduction = "wnn.umap", group.by = "manual.annotation", cols = WT.palette , pt.size = 1 )
```


```{r fig.width=8, fig.height=12, warning=FALSE, message=FALSE}
Idents(B6_WT)<-"RNA_snn_res.1.5"
B6_WT$RNA_snn_res.1.5 <- factor(B6_WT$RNA_snn_res.1.5, levels =c("14","7","4","2","6","1","10","3","5","11","0","8","13","9","12"))
genes <- c("Il2ra","adt_CD25","Ptcra","Top2a","Mki67","Cdk1","Rag1","Rag2","Cd4","adt_CD4","Cd8a","adt_CD8","Cd27","Cd69","adt_CD69","Cd5","Tox2","Ccr7","adt_CCR7","S1pr1","Sell","Cxcr3","Ccl5","Gzma","Nkg7","Klra1","Trdc","adt_TCRgd","Foxp3","Trp53inp1","Bmf","percent.mt")
dp <- DotPlot(B6_WT, features = genes) + geom_point(aes(size=pct.exp), shape = 21, colour="black", stroke=0.5) + scale_colour_gradient2(low = "steelblue", mid = "ivory1", high = "red") + guides(size=guide_legend(override.aes=list(shape=21, colour="black", fill="white"))) + RotatedAxis() + coord_flip()
dp + labs(title = "Expression of marker genes and surface proteins in clusters of BALB/c Wild Type")
```
# B6 (WT) subset
```{r}
DimPlot(B6_WT, group.by = "RNA_snn_res.1.5",reduction = "umap" , pt.size = 0.2, cols = paletteer::paletteer_d("ggthemes::Hue_Circle"), label = TRUE)
```
# B6 (WT) cluster annotation
```{r}
mananno.palette <- rev(c("#9E0142FF","#D53E4FFF","#F46D43FF","#FDAE61FF","#FEE08BFF","#ABDDA4FF","#66C2A5FF","#3288BDFF")) 
names(mananno.palette) <- c("DN3a","DN3b/ISP/DP blast","DPsm","DP69+","SP4 immature", "SP4 mature", "NKT", "Tgd")
B6_WT$manual.annotation <- factor(B6_WT$manual.annotation, levels = c("DN3a","DN3b/ISP/DP blast","DPsm","DP69+","SP4 immature", "SP4 mature", "NKT", "Tgd"))
DimPlot(B6_WT,reduction = "umap", group.by = "manual.annotation", label = T,pt.size = 0.2,label.size = 4, cols = mananno.palette)
```

```{r}
load(file= "/mnt/NASBIOINFO_CP/LALT/BIOINFO/PTEN/03_Analysis/Analysis2024/RObject/B6/B6WT_scigenex20.3.Robj")
load(file= "/mnt/NASBIOINFO_CP/LALT/BIOINFO/PTEN/03_Analysis/Analysis2024/RObject/BC/miceWT_scigenex20.3.Robj")
BCWT_scigenex20.3 <- miceWT_scigenex20.3
rm(miceWT_scigenex20.3)
```

```{r fig.width=12, fig.height=12, warning=FALSE, message=FALSE}
##Heatmap of representative genes
B6WT_scigenex20.3 <- top_genes(B6WT_scigenex20.3)
#By manual annotation
Idents(B6_WT)<-"manual.annotation"
B6_WT$manual.annotation <- factor(B6_WT$manual.annotation, levels = c("DN3a","DN3b/ISP/DP blast","DPsm","DP69+","SP4 immature", "SP4 mature", "NKT", "Tgd"))
plot_ggheatmap(B6WT_scigenex20.3, 
               use_top_genes = FALSE,
               ident=Idents(B6_WT),
               hide_gene_name = T) + ggtitle("Heatmap of Co-Expressed Gene Modules Identified by SciGeneX (k = 20, inflation= 3) grouped by cell types") +theme(strip.text.y = element_text(size=4))
```



```{r}
# Function to calculate Jaccard Index
jaccard_index <- function(set1, set2) {
  length(intersect(set1, set2)) / length(union(set1, set2))
}
```

```{r fig.width=12, fig.height=6, warning=FALSE, message=FALSE}
BCWT_modules <- BCWT_scigenex20.3@gene_clusters 
B6WT_modules <- B6WT_scigenex20.3@gene_clusters 
# Initialize a matrix to store the Jaccard Indices
jaccard_matrix <- matrix(nrow = length(B6WT_modules), ncol = length(BCWT_modules),
                         dimnames = list(names(B6WT_modules), names(BCWT_modules)))

# Nested loops to compute Jaccard Index for each pair
for (i in seq_along(B6WT_modules)) {
  for (j in seq_along(BCWT_modules)) {
    jaccard_matrix[i, j] <- jaccard_index(B6WT_modules[[i]], BCWT_modules[[j]])
  }
}

jaccard_df <- as.data.frame.table(jaccard_matrix)
names(jaccard_df) <- c("B6WT_Module", "BCWT_Module", "Jaccard_Index")
# Generating the plot
heatmap_plot <- ggplot(jaccard_df, aes(x = factor(B6WT_Module, levels = unique(jaccard_df$B6WT_Module)), 
                                       y = factor(BCWT_Module, levels = unique(jaccard_df$BCWT_Module)), 
                                       fill = Jaccard_Index)) +
  geom_tile() +  
  scale_fill_gradientn(colours = (viridis(4))) +  
  scale_x_discrete(position = "bottom") + 
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),  
        axis.title.x = element_blank(),  
        axis.title.y = element_blank(),  
        plot.title = element_text(hjust = 0.5),  
        legend.position = "right", 
        legend.key.width = unit(1, "cm"), 
        plot.margin = margin(t = 1, r = 1, b = 1, l = 1, unit = "cm")) + 
  xlab("B6 Modules") +  
  ylab("BC Modules") +  
  ggtitle("Heatmap of Jaccard Indices Between Gene Modules from BC-OTII (WT) and B6-OTII (WT)") 

# Afficher le tracé
print(heatmap_plot)
```


## Cell type annotation of BC mice
```{r fig.width=6, fig.height=10, warning=FALSE, message=FALSE}
DotPlot(BC_WT, features = c("auc_8", "adt_CD25", "auc_2", "auc_3", "auc_6", "adt_CD4", "adt_CD8", "adt_CD69", "adt_CCR7", "auc_13", "adt_TCRgd", "auc_5", "auc_9", "auc_12", "auc_11", "percent.mt", "Myc"), group.by = "manual.annotation") +
  geom_point(aes(size = pct.exp), shape = 21, colour = "black", stroke = 0.5) +
  scale_colour_gradient2(low = "steelblue", mid = "ivory1", high = "red") +
  guides(size = guide_legend(override.aes = list(shape = 21, colour = "black", fill = "white"))) +
  coord_flip() +
 theme(
    legend.position = "left",  # Moves the legend to the left
    axis.title.y = element_blank(), 
    axis.line.y.right = element_line(linewidth = 1, color = "black"),# Removes the default y-axis title
    axis.text.x = element_text(angle = 0, hjust = 1)  # Ensures x-axis labels are properly aligned
  ) + RotatedAxis() + coord_flip() + labs(title = "Expression of gene modules (WT) and surface proteins by clusters")

```

```{r fig.width=16, fig.height=6, warning=FALSE, message=FALSE}
bc.combined.cc$wsnn_res.2 <- factor(bc.combined.cc$wsnn_res.2, levels=c("26", "31", "10", "13", "0", "16", "29", "33", "6", "17", "9", "23", "14", "12", "2", "1", "20", "28", "35", "4", "8", "43", "19", "38", "39", "5", "22", "18", "37", "30", "34", "25", "32", "42", "11", "21", "24", "27", "3", "36", "40", "7", "41", "15"))
DotPlot(bc.combined.cc, features = c("auc_8","adt_CD25","auc_2","auc_3","auc_6","adt_CD4","adt_CD8","adt_CD69","adt_CCR7","auc_13","adt_TCRgd","auc_5","auc_10","auc_12","auc_11","percent.mt", "Myc"), group.by = "wsnn_res.2", col.min = -2.5) + geom_point(aes(size=pct.exp), shape = 21, colour="black", stroke=0.5) + scale_colour_gradient2(low = "steelblue", mid = "ivory1", high = "red") + guides(size=guide_legend(override.aes=list(shape=21, colour="black", fill="white"))) + RotatedAxis()+ coord_flip()+ labs(title = "Expression of gene modules (WT) and surface proteins by clusters")
```

```{r fig.width=12, fig.height=6, warning=FALSE, message=FALSE}
FeaturePlot(bc.combined.cc, features = c("auc_8", "adt_CD25", "auc_2", "auc_3", "auc_6", "adt_CD4", "adt_CD8", "adt_CD69", "adt_CCR7", "auc_10","auc_13", "adt_TCRgd", "auc_5", "auc_9", "auc_12", "auc_11", "percent.mt", "Myc"), reduction = "wnn.umap", ncol = 5)& scale_colour_gradientn(colours = rev(viridis::rocket(10))) &
    NoLegend() + 
    theme_void()+
    theme(plot.title = element_text(hjust = 0.5))
```

# UMAP Visualization of BC clusters
```{r}
umap.palette <- c("#C8102EFF","#FF6720FF","#FFC72CFF","#F9D14AFF","#009739FF","#82C45FFF","#318F49FF","#AFDE62FF","#3CBC38FF","#1D4289FF","#0032A0FF","#702F8AFF","#FFCD00FF","#E4002BFF","#B50200FF","#FF6720FF","#94B594FF","#AC145AFF","#42511AFF","#889D35FF","#007E2FFF","#D3D175FF","#DA6C41FF","#AA361DFF","#0CB4BBFF","#00B7A7","#ED968CFF","#381A61FF","#473D7DFF","#B75347FF","#DF7E66FF","#E09351FF","#EDC775FF","#E78429FF","#224B5EFF","#2673A3FF","#16317DFF","#FFCD12FF","#B86092FF","#721B3EFF","#88A0DCFF","#7C4B73FF","#AB3329FF","#A40000FF")
names(umap.palette) <- c("1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44")
DimPlot(bc.combined.cc, group.by = "wsnn_res.2", reduction = "wnn.umap", cols = umap.palette,label =T, repel =TRUE, label.size = 4 ,label.color = "black", pt.size = 0.1)
```



# BC VlnPlot - Visualization the expression of gene modules  and ADT in BC mice
```{r fig.width=14, fig.height=8, warning=TRUE}
## Modules gene
p8 <- VlnPlot(bc.combined.cc, features = "auc_8", group.by = "wsnn_res.2", pt.size = 0) +geom_hline(yintercept = 0.07, linetype ="dashed", linewidth = 1, color="black")+ theme(legend.position = "none")+
   stat_summary(fun = "median",
            geom = "crossbar",
            color = 'firebrick1', show.legend = F)
p2 <- VlnPlot(bc.combined.cc, features = "auc_2", group.by = "wsnn_res.2", pt.size = 0) +geom_hline(yintercept = 0.055, linetype ="dashed", linewidth = 1, color="black")+ theme(legend.position = "none")+
   stat_summary(fun = "median",
            geom = "crossbar",
            color = 'firebrick1', show.legend = F)
p3 <- VlnPlot(bc.combined.cc, features = "auc_3", group.by = "wsnn_res.2", pt.size = 0) +geom_hline(yintercept = 0.1, linetype ="dashed", linewidth = 1, color="black")+ theme(legend.position = "none")+
   stat_summary(fun = "median",
            geom = "crossbar",
            color = 'firebrick1', show.legend = F)
p6 <- VlnPlot(bc.combined.cc, features = "auc_6", group.by = "wsnn_res.2", pt.size = 0) +geom_hline(yintercept = 0.2, linetype ="dashed", linewidth = 1, color="black")+ theme(legend.position = "none")+
   stat_summary(fun = "median",
            geom = "crossbar",
            color = 'firebrick1', show.legend = F)
p13 <- VlnPlot(bc.combined.cc, features = "auc_13", group.by = "wsnn_res.2", pt.size = 0) +geom_hline(yintercept = 0.1, linetype ="dashed", linewidth = 1, color="black")+ theme(legend.position = "none")+
   stat_summary(fun = "median",
            geom = "crossbar",
            color = 'firebrick1', show.legend = F)
p5 <- VlnPlot(bc.combined.cc, features = "auc_5", group.by = "wsnn_res.2", pt.size = 0) +geom_hline(yintercept = 0.1, linetype ="dashed", linewidth = 1, color="black")+ theme(legend.position = "none")+
   stat_summary(fun = "median",
            geom = "crossbar",
            color = 'firebrick1', show.legend = F)
p9 <- VlnPlot(bc.combined.cc, features = "auc_9", group.by = "wsnn_res.2", pt.size = 0) +geom_hline(yintercept = 0.2, linetype ="dashed", linewidth = 1, color="black")+ theme(legend.position = "none")+
   stat_summary(fun = "median",
            geom = "crossbar",
            color = 'firebrick1', show.legend = F)
p12 <- VlnPlot(bc.combined.cc, features = "auc_12", group.by = "wsnn_res.2", pt.size = 0) +geom_hline(yintercept = 0.075, linetype ="dashed", linewidth = 1, color="black")+ theme(legend.position = "none")+
   stat_summary(fun = "median",
            geom = "crossbar",
            color = 'firebrick1', show.legend = F)
p11 <- VlnPlot(bc.combined.cc, features = "auc_11", group.by = "wsnn_res.2", pt.size = 0) +geom_hline(yintercept = 0.5, linetype ="dashed", linewidth = 1, color="black")+ theme(legend.position = "none")+
   stat_summary(fun = "median",
            geom = "crossbar",
            color = 'firebrick1', show.legend = F)
plot_grid(p8, p2, p3,p6 ,ncol = 2)
plot_grid(p13,p5,p9,p12,p11, ncol = 2)

#ADT

p.Cd25 <- VlnPlot(bc.combined.cc, features = "adt_CD25", group.by = "wsnn_res.2", pt.size = 0)+geom_hline(yintercept = 0.8, linetype ="dashed", linewidth = 1, color="black")+ theme(legend.position = "none")+
  stat_summary(fun = "median",
           geom = "crossbar",
           color = 'firebrick1', show.legend = F)

p.Cd8 <- VlnPlot(bc.combined.cc, features = "adt_CD8", group.by = "wsnn_res.2", pt.size = 0)+geom_hline(yintercept = 1.9, linetype ="dashed", linewidth = 1, color="black")+ theme(legend.position = "none")+
  stat_summary(fun = "median",
           geom = "crossbar",
           color = 'firebrick1', show.legend = F)

p.Cd4 <- VlnPlot(bc.combined.cc, features = "adt_CD4", group.by = "wsnn_res.2", pt.size = 0)+geom_hline(yintercept = 1.8, linetype ="dashed", linewidth = 1, color="black")+ theme(legend.position = "none")+
  stat_summary(fun = "median",
           geom = "crossbar",
           color = 'firebrick1', show.legend = F)

p.Cd3 <- VlnPlot(bc.combined.cc, features = "adt_CD3", group.by = "wsnn_res.2", pt.size = 0)+geom_hline(yintercept = 0.5, linetype ="dashed", linewidth = 1, color="black")+ theme(legend.position = "none")+
  stat_summary(fun = "median",
           geom = "crossbar",
           color = 'firebrick1', show.legend = F)

p.Tgd <- VlnPlot(bc.combined.cc, features = "adt_TCRgd", group.by = "wsnn_res.2", pt.size = 0)+geom_hline(yintercept = 0.5, linetype ="dashed", linewidth = 1, color="black")+ theme(legend.position = "none")+
  stat_summary(fun = "median",
           geom = "crossbar",
           color = 'firebrick1', show.legend = F)

p.Cd69 <- VlnPlot(bc.combined.cc, features = "adt_CD69", group.by = "wsnn_res.2", pt.size = 0)+geom_hline(yintercept = 0.04, linetype ="dashed", linewidth = 1, color="black")+theme(legend.position = "none")+
  stat_summary(fun = "median",
           geom = "crossbar",
           color = 'firebrick1', show.legend = F)

p.Ccr7 <- VlnPlot(bc.combined.cc, features = "adt_CCR7", group.by = "wsnn_res.2", pt.size = 0)+geom_hline(yintercept = 0.04, linetype ="dashed", linewidth = 1, color="black")+theme(legend.position = "none")+
  stat_summary(fun = "median",
           geom = "crossbar",
           color = 'firebrick1', show.legend = F)

#Myc 
p.Myc <- VlnPlot(bc.combined.cc, features = "Myc", group.by = "wsnn_res.2", pt.size = 0) +geom_hline(yintercept = 0.2, linetype ="dashed", linewidth = 1, color="black")+theme(legend.position = "none")+
  stat_summary(fun = "median",
           geom = "crossbar",
           color = 'firebrick1', show.legend = F)

plot_grid(p.Cd25,p.Cd4,p.Cd8, p.Cd3, ncol = 2)
plot_grid(p.Cd69,p.Ccr7,p.Tgd, p.Myc, ncol = 2)
```



# UMAP Visualization of cell annotations
```{r fig.width=10, fig.height=8, warning=TRUE}
#Palettes for "sig.annotation" of BC
palette1 <- c("#F7FCB9FF", "#E6F598FF", "#D9F0A3FF", "#ADDD8EFF","#82C45FFF","#78C679FF", "#238443FF", "#41AB5DFF","#007E2FFF","#006837FF", "#004529FF")
names(palette1) <- c("DN","DPblast","DPsm","DPsmCD25+","DPsmCCR7+","DP69","DP to SP","SP8","SP8CD69+","SP4","SP4CD69+")
#palette2 <- c("#F94A17","#E22611","#E4002BFF")
palette2 <- c("#F94A17","#DA5A5AFF","#E4002BFF")

names(palette2) <- c("DPblast Myc+","DPsm Myc+","DP69 Myc+")
palette3 <- c("#88A0DCFF","#94B594FF","#889D35FF","#42511AFF")
names(palette3) <- c("NK","Tgd early","Tgd mature","MAIT")
palette4 <- c("#C71000","#E3B1D6")
names(palette4) <- c("Undefined Myc+","Dying cells")
bc.combined.cc$sig.annotation <- factor(bc.combined.cc$sig.annotation, levels = c("DN","DPblast","DPsm","DPsmCD25+","DPsmCCR7+","DP69","DP to SP","SP8","SP8CD69+","SP4","SP4CD69+","Tgd early","Tgd mature","MAIT","NK","DPblast Myc+","DPsm Myc+","DP69 Myc+","Undefined Myc+","Dying cells", "unknown"))
DimPlot(bc.combined.cc, group.by = "sig.annotation", reduction = "wnn.umap",cols = c(palette1,palette2, palette3, palette4), pt.size = 0.1, label = FALSE, repel = T, label.size = 4, label.color = "#0D0C0BFF")
```
# UMAP Visualization of PTENdel mice, control mice and both
```{r fig.width=5, fig.height=5, warning=TRUE}
Idents(bc.combined.cc) <- "miceID"
Ptendel_bcsubset <- subset(bc.combined.cc,idents = c("47","85","84","39","52","70","80","53","69"))
miceWT_bcsubset <- subset(bc.combined.cc,idents = c("51","83"))
palette.WT <- c(rep("#009739FF",times = 11))
palette.tumor <- c(rep("#E4002BFF", times = 4))
palette.others <- c(rep("#88A0DCFF",times = 4),"#AB3329FF")
names(palette.WT) <- c("DN","DPblast","DPsm","DPsmCD25+","DPsmCCR7+","DP69","DP to SP","SP8","SP8CD69+","SP4","SP4CD69+")
names(palette.tumor) <- c("DPblast Myc+","DPsm Myc+","DP69 Myc+", "Undefined Myc+")
names(palette.others) <- c("NK","Tgd early","Tgd mature","MAIT","Dying cells")
DimPlot(Ptendel_bcsubset,group.by = "sig.annotation", reduction = "wnn.umap", cols =c(palette.WT,palette.tumor,palette.others), pt.size = 0.5, repel =TRUE)&
    NoLegend() + 
    theme_void()+
    theme(plot.title = element_text(hjust = 0.5)) 
DimPlot(miceWT_bcsubset,group.by = "sig.annotation", reduction = "wnn.umap", cols = c(palette.WT,palette.tumor,palette.others), pt.size = 0.5, repel =TRUE)&
    NoLegend() + 
    theme_void()+
    theme(plot.title = element_text(hjust = 0.5)) 
DimPlot(bc.combined.cc,group.by = "sig.annotation", reduction = "wnn.umap", cols = c(palette.WT,palette.tumor,palette.others), pt.size = 0.5, repel =TRUE)&
    NoLegend() + 
    theme_void()+
    theme(plot.title = element_text(hjust = 0.5))

```

# Distribution of cell types in Thymus BC
```{r fig.width=6, fig.height=5, warning=TRUE}
Idents(bc.combined.cc) <- "annotation"
Thymus <- subset(bc.combined.cc, idents = c("Thymus-WT","Thymus-PreTum late","Thymus-Tum"))
Idents(Thymus) <- "sig.annotation"
T.bar <- subset(Thymus,  idents = c("DN","DPblast","DPsm","DPsmCD25+","DPsmCCR7+","DP69","SP8","SP4","SP4CD69+","Tgd early","Tgd mature","MAIT","NK","DPblast Myc+","DPsm Myc+","DP69 Myc+","Undefined Myc+","Dying cells", "unknown"))

data1 <- as.data.frame(prop.table(table(T.bar@meta.data$sig.annotation,T.bar@meta.data$annotation),2)*100)
data1 <-na.omit(data1)
#order leukestage level
data1$Var2 <- factor(data1$Var2, levels = c("Thymus-WT","Thymus-PreTum late","Thymus-Tum"))
cellnumber <- data.frame(colSums(table(T.bar@meta.data$MULTI_ID,T.bar@meta.data$annotation)))
cellnumber$annotation <- rownames(cellnumber)
row_order <-c("Thymus-WT","Thymus-PreTum late","Thymus-Tum")
cellnumber <- cellnumber[row_order,]
annotation <- data1$Var2
sig.annotation <- data1$Var1
sig.annotation <- factor(sig.annotation, levels =rev( c("DN","DPblast","DPsm","DPsmCD25+","DPsmCCR7+","DP69","SP8","SP4","SP4CD69+","Tgd early","Tgd mature","MAIT","NK","DPblast Myc+","DPsm Myc+","DP69 Myc+","Undefined Myc+","Dying cells", "unknown")))
Percentage <- data1$Freq
text <- cellnumber$colSums.table.T.bar.meta.data.MULTI_ID..T.bar.meta.data.annotation..
ggplot(data1, aes(fill=sig.annotation, y=Percentage, x=annotation)) + 
    geom_bar(position="stack", stat="identity", width=0.7)+ labs(title = "Distribution of cell types in Thymus")+
   xlab("Stade of leukemia")+ylab("Percentage")+ theme(legend.position="bottom")+scale_fill_manual(values =c(palette1,palette2,palette3,palette4)) +theme_light()+geom_hline(yintercept=c(25,50,75), linetype="dashed", color = "grey60")+ annotate("text", x = 1:3, y=103, label = c(text))
```

# Distribution of cell types in Spleen BC
```{r fig.width=6, fig.height=5, warning=TRUE}
Idents(bc.combined.cc) <- "annotation"
Spleen <- subset(bc.combined.cc, idents = c("Spleen-WT","Spleen-PreTum late","Spleen-Tum"))
Idents(Spleen) <- "sig.annotation"
maturebar <- subset(Spleen,  idents = c("SP8","SP4","SP4CD69+","Tgd early","Tgd mature","MAIT","NK","Undefined Myc+", "Dying cells"))

data2 <- as.data.frame(prop.table(table(maturebar@meta.data$sig.annotation,maturebar@meta.data$annotation),2)*100)
data2 <- na.omit(data2)
#order leukestage level
data2$Var2 <- factor(data2$Var2, levels = c("Spleen-WT","Spleen-PreTum late", "Spleen-Tum"))
cellnumber <- data.frame(colSums(table(maturebar@meta.data$MULTI_ID,maturebar@meta.data$annotation)))
cellnumber$annotation <- rownames(cellnumber)
row_order <-c("Spleen-WT","Spleen-PreTum late", "Spleen-Tum")
cellnumber <- cellnumber[row_order,]
annotation <- data2$Var2
sig.annotation <- data2$Var1
sig.annotation <- factor(sig.annotation, levels =rev( c("SP8","SP4","SP4CD69+","Tgd early","Tgd mature","MAIT","NK","Undefined Myc+", "Dying cells")))
Percentage <- data2$Freq
text <- cellnumber$colSums.table.maturebar.meta.data.MULTI_ID..maturebar.meta.data.annotation..
ggplot(data2, aes(fill=sig.annotation, y=Percentage, x=annotation)) + 
    geom_bar(position="stack", stat="identity", width=0.7)+
   xlab("Stade of leukemia")+ylab("Percentage")+ theme(legend.position="bottom")+scale_fill_manual(values =c(palette1,palette2,palette3,palette4)) +theme_light()+geom_hline(yintercept=c(25,50,75), linetype="dashed", color = "grey60")+ annotate("text", x = 1:3, y=103, label = c(text))+ labs(title = "Distribution of cell types in Spleen")
```


## Cell type annotation of B6 mice

```{r}
umap.palette <- c("#C8102EFF","#FF6720FF","#FFC72CFF","#F9D14AFF","#009739FF","#82C45FFF","#318F49FF","#AFDE62FF","#3CBC38FF","#1D4289FF","#0032A0FF","#702F8AFF","#FFCD00FF","#E4002BFF","#B50200FF","#FF6720FF","#94B594FF","#AC145AFF","#42511AFF","#889D35FF","#007E2FFF","#D3D175FF","#DA6C41FF","#AA361DFF","#0CB4BBFF","#00B7A7","#ED968CFF","#381A61FF","#473D7DFF","#B75347FF","#DF7E66FF","#E09351FF","#EDC775FF","#E78429FF","#224B5EFF","#2673A3FF","#16317DFF","#FFCD12FF","#B86092FF","#721B3EFF","#88A0DCFF","#7C4B73FF","#AB3329FF","#A40000FF")

#names(umap.palette) <- c("1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44")
DimPlot(b6.combined.cc, group.by = "RNA_snn_res.1", reduction = "umap", cols = umap.palette,label =T, label.size = 4 ,label.color = "black", pt.size = 0.5, repel =TRUE)
```

```{r}
b6.combined.cc$RNA_snn_res.1 <- factor(b6.combined.cc$RNA_snn_res.1, levels=c("16","4","9","1","22","3","8","14","10","17","5","0","2","6","19","21","15","11","20","7","18","12","13","24","23"))
# DotPlot(b6.combined.cc, features = c("aucwt_11","adt_CD25","aucwt_5","adt_CD4","adt_CD8","aucwt_1","aucwt_2","aucwt_6","aucwt_4","aucwt_12","aucwt_13","aucwt_7","adt_CD69","adt_CCR7","adt_TCRgd","percent.mt", "Myc"), group.by = "RNA_snn_res.1", col.min = -2.5) + geom_point(aes(size=pct.exp), shape = 21, colour="black", stroke=0.5) + scale_colour_gradient2(low = "steelblue", mid = "ivory1", high = "red") + guides(size=guide_legend(override.aes=list(shape=21, colour="black", fill="white"))) + RotatedAxis()+ coord_flip()+ labs(title = "Expression of gene modules (WT) and surface proteins by clusters")
```

```{r fig.width=12, fig.height=6, warning=FALSE, message=FALSE}
FeaturePlot(b6.combined.cc, features = c("aucwt_11","adt_CD25","aucwt_5","adt_CD4","adt_CD8","aucwt_1","aucwt_2","aucwt_6","aucwt_4","aucwt_12","aucwt_13","aucwt_7","adt_CD69","adt_CCR7","adt_TCRgd","percent.mt", "Myc"), reduction = "umap", ncol = 5)& scale_colour_gradientn(colours = rev(viridis::rocket(10))) &
    NoLegend() + 
    theme_void()+
    theme(plot.title = element_text(hjust = 0.5))
```

# B6 VlnPlot - Visualization the expression of gene modules and ADT in B6 mice
```{r fig.width=14, fig.height=8, warning=TRUE}

## Modules gene
p11 <- VlnPlot(b6.combined.cc, features = c("aucwt_11"), group.by = "RNA_snn_res.1", pt.size = 0)+ theme(legend.position = "none")&geom_hline(yintercept = 0.035, linetype ="dashed", size = 1, color="black")&
  stat_summary(fun = "median",
           geom = "crossbar",
           color = 'firebrick1', 
           show.legend = F)
p5 <- VlnPlot(b6.combined.cc, features = "aucwt_5", group.by = "RNA_snn_res.1", pt.size = 0) + theme(legend.position = "none")+geom_hline(yintercept = 0.16, linetype ="dashed", size = 1, color="black")+
  stat_summary(fun = "median",
           geom = "crossbar",
           color = 'firebrick1', show.legend = F)
p1 <- VlnPlot(b6.combined.cc, features = "aucwt_1", group.by = "RNA_snn_res.1", pt.size = 0) + theme(legend.position = "none")+geom_hline(yintercept = 0.087, linetype ="dashed", size = 1, color="black")+
  stat_summary(fun = "median",
           geom = "crossbar",
           color = 'firebrick1', show.legend = F)
p2 <- VlnPlot(b6.combined.cc, features = "aucwt_2", group.by = "RNA_snn_res.1", pt.size = 0) + theme(legend.position = "none")+geom_hline(yintercept = 0.12, linetype ="dashed", size = 1, color="black")+
  stat_summary(fun = "median",
           geom = "crossbar",
           color = 'firebrick1', show.legend = F)

p12 <- VlnPlot(b6.combined.cc, features = "aucwt_12", group.by = "RNA_snn_res.1", pt.size = 0)+ theme(legend.position = "none") +geom_hline(yintercept = 0.082, linetype ="dashed", size = 1, color="black")+
  stat_summary(fun = "median",
           geom = "crossbar",
           color = 'firebrick1', show.legend = F)
p13 <- VlnPlot(b6.combined.cc, features = "aucwt_13", group.by = "RNA_snn_res.1", pt.size = 0)+ theme(legend.position = "none") +geom_hline(yintercept = 0.049, linetype ="dashed", size = 1, color="black")+
  stat_summary(fun = "median",
           geom = "crossbar",
           color = 'firebrick1', show.legend = F)

p7 <- VlnPlot(b6.combined.cc, features = "aucwt_7", group.by = "RNA_snn_res.1", pt.size = 0) + theme(legend.position = "none")+geom_hline(yintercept = 0.04, linetype ="dashed", size = 1, color="black")+
  stat_summary(fun = "median",
           geom = "crossbar",
           color = 'firebrick1', show.legend = F)
p6 <- VlnPlot(b6.combined.cc, features = "aucwt_6", group.by ="RNA_snn_res.1", pt.size = 0) + theme(legend.position = "none")+geom_hline(yintercept = 0.41, linetype ="dashed", size = 1, color="black")+
  stat_summary(fun = "median",
           geom = "crossbar",
           color = 'firebrick1', show.legend = F)
p4 <- VlnPlot(b6.combined.cc, features = "aucwt_4", group.by = "RNA_snn_res.1", pt.size = 0)+ theme(legend.position = "none") +geom_hline(yintercept = 0.16, linetype ="dashed", size = 1, color="black")+
  stat_summary(fun = "median",
           geom = "crossbar",
           color = 'firebrick1', show.legend = F)

p.Myc <- VlnPlot(b6.combined.cc, features = "Myc", group.by = "RNA_snn_res.1", pt.size = 0) + theme(legend.position = "none")+geom_hline(yintercept = 0.5, linetype ="dashed", size = 1, color="black")+
  stat_summary(fun = "median",
           geom = "crossbar",
           color = 'firebrick1', show.legend = F)
# adt_CD8
p.Cd8 <- VlnPlot(b6.combined.cc, features = "adt_CD8", group.by = "RNA_snn_res.1", pt.size = 0)+ theme(legend.position = "none") +geom_hline(yintercept = 2, linetype ="dashed", size = 1, color="black")+
  stat_summary(fun = "median",
           geom = "crossbar",
           color = 'firebrick1', show.legend = F)
# adt_CD4 
p.Cd4 <- VlnPlot(b6.combined.cc, features = "adt_CD4", group.by = "RNA_snn_res.1",pt.size = 0)+ theme(legend.position = "none") +geom_hline(yintercept = 1, linetype ="dashed", size = 1, color="black")+
   stat_summary(fun = "median",
            geom = "crossbar",
            color = 'firebrick1', show.legend = F)
p.Cd25 <- VlnPlot(b6.combined.cc, features = "adt_CD25", group.by = "RNA_snn_res.1", pt.size = 0)+ theme(legend.position = "none") +geom_hline(yintercept = 1, linetype ="dashed", size = 1, color="black")+
   stat_summary(fun = "median",
            geom = "crossbar",
            color = 'firebrick1', show.legend = F)

plot_grid(p11,p.Cd25, p5, p.Cd4, p.Cd8, p1, p2,p4,p6, ncol = 3)
plot_grid(p12, p13, p7, p.Myc, ncol = 2)

```


```{r fig.width=10, fig.height=10, warning=TRUE}
# Dotplot of the signatures and ADT used for annotation 
Idents(b6.combined.cc)<-"RNA_snn_res.1"
b6.combined.cc$RNA_snn_res.1 <- factor(b6.combined.cc$RNA_snn_res.1, levels =c("16","1","3","22","10","4","9","8","17","14","5","0","2","6","19","21","15","20","7","12","24","13","11","18","23"))
genes <- c("Il2ra","adt_CD25","Ptcra","Top2a","Mki67","Cdk1","Rag1","Rag2","Cd4","adt_CD4","Cd8a","adt_CD8","Cd27","Cd69","adt_CD69","Cd5","Tox2","Ccr7","adt_CCR7","S1pr1","Sell","Cxcr3","Ccl5","Gzma","Nkg7","Klra1","Trdc","adt_TCRgd","Foxp3","Trp53inp1","Bmf","percent.mt","Myc")
DotPlot(b6.combined.cc, features = genes) + geom_point(aes(size=pct.exp), shape = 21, colour="black", stroke=0.5) + scale_colour_gradient2(low = "steelblue", mid = "ivory1", high = "red") + guides(size=guide_legend(override.aes=list(shape=21, colour="black", fill="white"))) + RotatedAxis() + coord_flip()+ labs(title = "Expression of marker genes and surface proteins in clusters of B6 mice")
```


```{r fig.width=12, fig.height=6, warning=FALSE, message=FALSE}
FeaturePlot(b6.combined.cc, features = c("aucwt_11","adt_CD25","aucwt_5","adt_CD4","adt_CD8","aucwt_1","aucwt_2","aucwt_6","aucwt_4","aucwt_12","aucwt_13","aucwt_7","adt_CD69","adt_CCR7","adt_TCRgd","percent.mt", "Myc"), reduction = "umap", ncol = 5)& scale_colour_gradientn(colours = rev(viridis::rocket(10))) &
    NoLegend() + 
    theme_void()+
    theme(plot.title = element_text(hjust = 0.5))
```

```{r}
b6.combined.cc$sig.annotation <- factor(b6.combined.cc$sig.annotation, levels = c("DN-early DP","DPsm","SP4 immature", "SP4 mature", "DPblast-Myc+","DPsm-Myc+","DP69-Myc+","SP4-Myc+","Undefined Myc+","Tgd/NK/NKT","Tgd","NK"))
# Palette colors for "sig.annotation" in B6
palette1 <- c("#F7FCB9FF","#D9F0A3FF","#5E9432FF","#3B7D31FF")
names(palette1) <- c("DN-early DP","DPsm","SP4 immature", "SP4 mature")
palette2 <- c("#F94A17","#DA5A5AFF","#E4002BFF","#C0392BFF")
names(palette2) <- c("DPblast-Myc+","DPsm-Myc+","DP69-Myc+", "SP4-Myc+")
palette3 <- c("#019875FF","#99B898FF","#88A0DCFF")
names(palette3) <- c("Tgd/NK/NKT","Tgd","NK")
palette4 <- c("#C71000")
names(palette4) <- c("Undefined Myc+")
DimPlot(b6.combined.cc, group.by = "sig.annotation", reduction = "umap", cols = c(palette1, palette2, palette3, palette4),label =F, label.size = 4 ,label.color = "black", pt.size = 1, repel =TRUE) &
    NoLegend() + 
    theme_void()+
    theme(plot.title = element_text(hjust = 0.5))
```


```{r fig.width=6, fig.height=5, warning=TRUE}
Idents(b6.combined.cc) <- "stage"
Thymusb6 <- subset(b6.combined.cc, idents = c("Thymus-WT","Thymus-PreTum","Thymus-Tum"))
Idents(Thymusb6) <- "sig.annotation"
TB6.bar <- subset(Thymusb6,  idents = c("DN-early DP","DPsm","SP4 immature", "SP4 mature", "DPblast-Myc+","DPsm-Myc+","DP69-Myc+","SP4-Myc+","Undefined Myc+","Tgd/NK/NKT","Tgd","NK"))

data1 <- as.data.frame(prop.table(table(TB6.bar@meta.data$sig.annotation,TB6.bar@meta.data$stage),2)*100)
data1 <-na.omit(data1)
#order leukestage level
data1$Var2 <- factor(data1$Var2, levels = c("Thymus-WT","Thymus-PreTum","Thymus-Tum"))
cellnumber <- data.frame(colSums(table(TB6.bar@meta.data$MULTI_ID,TB6.bar@meta.data$stage)))
cellnumber$stage <- rownames(cellnumber)
row_order <-c("Thymus-WT","Thymus-PreTum","Thymus-Tum")
cellnumber <- cellnumber[row_order,]
stage <- data1$Var2
sig.annotation <- data1$Var1
sig.annotation <- factor(sig.annotation, levels =rev(c("DN-early DP","DPsm","SP4 immature", "SP4 mature", "DPblast-Myc+","DPsm-Myc+","DP69-Myc+","SP4-Myc+","Undefined Myc+","Tgd/NK/NKT","Tgd","NK")))
Percentage <- data1$Freq
text <- cellnumber$colSums.table.TB6.bar.meta.data.MULTI_ID..TB6.bar.meta.data.stage..
ggplot(data1, aes(fill=sig.annotation, y=Percentage, x=stage)) + 
    geom_bar(position="stack", stat="identity", width=0.7)+ labs(title = "Distribution of cell types in Thymus")+
   xlab("Stade of leukemia")+ylab("Percentage")+ theme(legend.position="bottom")+scale_fill_manual(values =c(palette1,palette2,palette3,palette4)) +theme_light()+geom_hline(yintercept=c(25,50,75), linetype="dashed", color = "grey60")+ annotate("text", x = 1:3, y=103, label = c(text))
```

```{r fig.width=6, fig.height=5, warning=TRUE}
Idents(b6.combined.cc) <- "stage"
Spleenb6 <- subset(b6.combined.cc, idents = c("Spleen-WT","Spleen-PreTum","Spleen-Tum"))
Idents(Spleenb6) <- "sig.annotation"
SB6.bar <- subset(Spleenb6,  idents = c("SP4 mature","DPblast-Myc+","DPsm-Myc+", "SP4-Myc+","Undefined Myc+","Tgd/NK/NKT","NK"))

data1 <- as.data.frame(prop.table(table(SB6.bar@meta.data$sig.annotation,SB6.bar@meta.data$stage),2)*100)
data1 <-na.omit(data1)
#order leukestage level
data1$Var2 <- factor(data1$Var2, levels = c("Spleen-WT","Spleen-PreTum","Spleen-Tum"))
cellnumber <- data.frame(colSums(table(SB6.bar@meta.data$MULTI_ID,SB6.bar@meta.data$stage)))
cellnumber$stage <- rownames(cellnumber)
row_order <-c("Spleen-WT","Spleen-PreTum","Spleen-Tum")
cellnumber <- cellnumber[row_order,]
stage <- data1$Var2
sig.annotation <- data1$Var1
sig.annotation <- factor(sig.annotation, levels =rev( c("SP4 mature","DPblast-Myc+","DPsm-Myc+", "SP4-Myc+","Undefined Myc+","Tgd/NK/NKT","NK")))
Percentage <- data1$Freq
text <- cellnumber$colSums.table.SB6.bar.meta.data.MULTI_ID..SB6.bar.meta.data.stage..
ggplot(data1, aes(fill=sig.annotation, y=Percentage, x=stage)) + 
    geom_bar(position="stack", stat="identity", width=0.7)+ labs(title = "Distribution of cell types in Spleen")+
   xlab("Stade of leukemia")+ylab("Percentage")+ theme(legend.position="bottom")+scale_fill_manual(values =c(palette1,palette2,palette3,palette4)) +theme_light()+geom_hline(yintercept=c(25,50,75), linetype="dashed", color = "grey60")+ annotate("text", x = 1:3, y=103, label = c(text))
```



# TCR analysis 
```{r}
OTIIpalette <- c("#7AD151FF","#414487FF","#FDE725FF","#DE4968FF","#808080FF")
names(OTIIpalette) <- c("OTII","OTII - incomplete","OTII - multichains","non OTII","unknown")
```

```{r fig.width=10, fig.height=4, warning=TRUE}
Idents(b6.combined.cc) <- "OTIIstatus"
p1 <- DimPlot(b6.combined.cc, group.by = "OTIIstatus",reduction = "umap" , cells.highlight = WhichCells(b6.combined.cc, idents = "OTII"),sizes.highlight = 0.5, cols.highlight = "#7AD151FF") + labs(title = "OTII")&
    NoLegend() + 
    theme_void()+
    theme(plot.title = element_text(hjust = 0.5, face = "bold"))
p2 <- DimPlot(b6.combined.cc, group.by = "OTIIstatus",reduction = "umap" , cells.highlight = WhichCells(b6.combined.cc, idents = "OTII - multichains"),sizes.highlight = 0.2, cols.highlight = "#FDE725FF") + labs(title = "OTII - multichains")&
    NoLegend() + 
    theme_void()+
    theme(plot.title = element_text(hjust = 0.5,face = "bold"))
p3 <-DimPlot(b6.combined.cc, group.by = "OTIIstatus",reduction = "umap" , cells.highlight = WhichCells(b6.combined.cc, idents = "non OTII"),sizes.highlight = 0.2, cols.highlight = "#DE4968FF") + labs(title = "non OTII")&
    NoLegend() + 
    theme_void()+
    theme(plot.title = element_text(hjust = 0.5,face = "bold"))

plot_grid(p1, p2, p3, ncol = 3)
FeaturePlot(b6.combined.cc, features = c("adt_TCR-VA2","adt_TCR-VB5"),order = T, reduction = "umap", pt.size = 0.2, ncol = 2 ) & scale_colour_gradientn(colours = rev(viridis::rocket(10)))
FeaturePlot(bc.combined.cc, features = c("adt_TCR-B","adt_TCR-VB5"),order = T, reduction = "wnn.umap", pt.size = 0.2, ncol = 2 ) & scale_colour_gradientn(colours = rev(viridis::rocket(10)))



```

```{r, warning=FALSE}
TCRb.ratio <- b6.combined.cc@assays$ADT$data[which(rownames(b6.combined.cc@assays$ADT$data) == "TCR-VB5"),]/b6.combined.cc@assays$ADT$data[which(rownames(b6.combined.cc@assays$ADT$data) == "TCR-B"),]
b6.combined.cc <- AddMetaData(object = b6.combined.cc,metadata = TCRb.ratio, col.name = "TCRBratio")
Thymusb6 <- AddMetaData(object = Thymusb6,metadata = TCRb.ratio, col.name = "TCRBratio")
FeaturePlot(object = b6.combined.cc, features = "TCRBratio", shape.by = "OTIIstatus",order = T,max.cutoff = 4) & scale_colour_gradientn(colours = rev(viridis::rocket(10)))

VlnPlot(b6.combined.cc, features = "adt_TCR-B", group.by = "sig.annotation",y.max = 4, pt.size = 0)& geom_boxplot(width=0.1, fill="white")
VlnPlot(b6.combined.cc, features = "TCRBratio", group.by = "sig.annotation", pt.size = 0)& geom_boxplot(width=0.1, fill="white")
VlnPlot(Thymusb6, features = "TCRBratio", group.by = "stage", pt.size = 0, y.max =4, cols = c("#9AC4F8","#fac341","#BB0A21") )& geom_boxplot(width=0.1, fill="white")
```


```{r fig.width=10, fig.height=4, warning=TRUE}
Idents(bc.combined.cc) <- "OTIIstatus"
p1 <- DimPlot(bc.combined.cc, group.by = "OTIIstatus",reduction = "wnn.umap" , cells.highlight = WhichCells(bc.combined.cc, idents = "OTII"),sizes.highlight = 0.5, cols.highlight = "#7AD151FF") + labs(title = "OTII")&
    NoLegend() + 
    theme_void()+
    theme(plot.title = element_text(hjust = 0.5, face = "bold"))
p2 <- DimPlot(bc.combined.cc, group.by = "OTIIstatus",reduction = "wnn.umap" , cells.highlight = WhichCells(bc.combined.cc, idents = "OTII - multichains"),sizes.highlight = 0.2, cols.highlight = "#FDE725FF") + labs(title = "OTII - multichains")&
    NoLegend() + 
    theme_void()+
    theme(plot.title = element_text(hjust = 0.5,face = "bold"))
p3 <-DimPlot(bc.combined.cc, group.by = "OTIIstatus",reduction = "wnn.umap" , cells.highlight = WhichCells(bc.combined.cc, idents = "non OTII"),sizes.highlight = 0.2, cols.highlight = "#DE4968FF") + labs(title = "non OTII")&
    NoLegend() + 
    theme_void()+
    theme(plot.title = element_text(hjust = 0.5,face = "bold"))

plot_grid(p1, p2, p3, ncol = 3)
FeaturePlot(bc.combined.cc, features = c("adt_TCR-VA2","adt_TCR-VB5"),order = T, reduction = "wnn.umap", pt.size = 0.2, ncol = 2 ) & scale_colour_gradientn(colours = rev(viridis::rocket(10)))
FeaturePlot(bc.combined.cc, features = c("adt_TCR-VA2","adt_TCR-VB5"),order = T, reduction = "wnn.umap", pt.size = 0.2, ncol = 2 ) & scale_colour_gradientn(colours = rev(viridis::rocket(10)))
```
# Distribution of OTIIstatus in Thymus control, preleukemic, leukemic
```{r}
Idents(b6.combined.cc) <- "OTIIstatus"
OTIIb6 <- subset(b6.combined.cc, idents = c("OTII","OTII - multichains","non OTII"))# Remove 'unknown' and 'OTII- incomplete" because not informative
Idents(OTIIb6) <- "stage"
Thymusb6 <- subset(OTIIb6, idents = c("Thymus-WT","Thymus-PreTum","Thymus-Tum"))
data1 <- as.data.frame(prop.table(table(Thymusb6@meta.data$OTIIstatus,Thymusb6@meta.data$stage),2)*100)
data1 <-na.omit(data1)
#order leukestage level
data1$Var2 <- factor(data1$Var2, levels = c("Thymus-WT","Thymus-PreTum","Thymus-Tum"))
cellnumber <- data.frame(colSums(table(Thymusb6@meta.data$MULTI_ID,Thymusb6@meta.data$stage)))
cellnumber$stage <- rownames(cellnumber)
row_order <-c("Thymus-WT","Thymus-PreTum","Thymus-Tum")
cellnumber <- cellnumber[row_order,]
stage <- data1$Var2
OTII_status <- data1$Var1
OTII_status <- factor(OTII_status, levels = rev(c("OTII","OTII - multichains", "non OTII"))) 
Percentage <- data1$Freq
text <- cellnumber$colSums.table.Thymusb6.meta.data.MULTI_ID..Thymusb6.meta.data.stage..
ggplot(data1, aes(fill=OTII_status, y=Percentage, x=stage)) + 
    geom_bar(position="stack", stat="identity", width=0.7)+
   xlab("Stade of leukemia")+ylab("Percentage")+ theme(legend.position="bottom")+scale_fill_manual(values = OTIIpalette) +theme_light()+geom_hline(yintercept=c(25,50,75), linetype="dashed", color = "black")+ annotate("text", x = 1:3, y=103, label = c(text)) + labs(title = "Distribution of OTII status in Thymus B6 OTII ")

```
```{r fig.width=10, fig.height=4, warning=TRUE}
DimPlot(Thymusb6, group.by = "OTIIstatus", split.by = "stage", reduction = "umap" ,cols = OTIIpalette, pt.size = 0.5, ncol= 3) + labs(title = "OTII status")
```

# ```{r}
# Idents(b6.combined.cc) <- "stage"
# ThymusWTb6 <- subset(OTIIb6, idents = c("Thymus-WT"))
# df_radarchartWT<-as.data.frame(prop.table(table(ThymusWTb6@meta.data$sig.annotation,ThymusWTb6@meta.data$OTIIstatus),2)*100)
# colnames(df_radarchartWT) <- c( "Annotation","OTIIstatus","Proportion")
# 
# ThymusPreTumb6 <- subset(OTIIb6, idents = c("Thymus-PreTum"))
# df_radarchartPreTum<-as.data.frame(prop.table(table(ThymusPreTumb6@meta.data$sig.annotation,ThymusPreTumb6@meta.data$OTIIstatus),2)*100)
# colnames(df_radarchartPreTum) <- c( "Annotation","OTIIstatus","Proportion")
# 
# ThymusTumb6 <- subset(OTIIb6, idents = c("Thymus-Tum"))
# df_radarchartTum<-as.data.frame(prop.table(table(ThymusTumb6@meta.data$sig.annotation,ThymusTumb6@meta.data$OTIIstatus),2)*100)
# colnames(df_radarchartTum) <- c( "Annotation","OTIIstatus","Proportion")
# 
# # Charger les données
# df_list <- list(df_radarchartWT, df_radarchartPreTum, df_radarchartTum)
# df_combined <- NULL
#   for (i in seq_along(df_list)) {
#     df_i <- subset(df_list[[i]], OTIIstatus == "non OTII")
#       df_ti <- data.frame(t(df_i[,"Proportion"]))
#       colnames(df_ti) <- df_i$Annotation
#       # Ajouter chaque data frame transformé au data frame combiné
#       df_combined <- rbind(df_combined, setNames(df_ti, colnames(df_ti)))
#       col_names <- c(col_names, paste(colnames(df_ti), labels[i], sep=" - "))
#   }
# df_combined <- rbind(rep(60, 12), rep(0, 12),df_combined)[, 1:12]
# rownames(df_combined) <- c("Max","Min","Thymus-WT","Thymus-PreTum","Thymus-Tum")
# op <- par(mar = c(1, 1, 1, 1))  # Ajuster les marges
# radarchart(df_combined, cglcol="black", cglty=2, cglwd=1, vlcex=1,
#                pcol="#DE4968FF",pty = 20, plty=1:3, 
#                caxislabels=paste(seq(from=0, to=60, by=15), "%"),
#                axislabcol="black", axistype=4,
#                title=paste("non OTII"),
#                plwd=4)
# 
# # Ajouter une légende
# #legend("topright", legend=rownames(df_combined)[3:5], lty=linetypes, lwd=2, cex=0.8)
# node_labels <- colnames(df_combined)
# node_positions <- seq(0, 2 * pi, length.out = length(node_labels) + 1)[-1]
# node_r <- rep(60, length(node_labels))  # Distance pour chaque nœud
# text_pos_x <- cos(node_positions) * (node_r + 5)  # Ajuster la position x
# text_pos_y <- sin(node_positions) * (node_r + 5)  # Ajuster la position y
# text_labels <- colnames(df_combined)
# 
# # Tracer les étiquettes
# text(text_pos_x, text_pos_y, labels = text_labels, cex = 0.8)
# 
# par(op)
# ```



```{r}
Spleenb6 <- subset(OTIIb6, idents = c("Spleen-WT","Spleen-PreTum","Spleen-Tum"))
data1 <- as.data.frame(prop.table(table(Spleenb6@meta.data$OTIIstatus,Spleenb6@meta.data$stage),2)*100)
data1 <-na.omit(data1)
#order leukestage level
data1$Var2 <- factor(data1$Var2, levels = c("Spleen-WT","Spleen-PreTum","Spleen-Tum"))
cellnumber <- data.frame(colSums(table(Spleenb6@meta.data$MULTI_ID,Spleenb6@meta.data$stage)))
cellnumber$stage <- rownames(cellnumber)
row_order <-c("Spleen-WT","Spleen-PreTum","Spleen-Tum")
cellnumber <- cellnumber[row_order,]
stage <- data1$Var2
OTII_status <- data1$Var1
OTII_status <- factor(OTII_status, levels = rev(c("OTII","OTII - multichains", "non OTII"))) 
Percentage <- data1$Freq
text <- cellnumber$colSums.table.Spleenb6.meta.data.MULTI_ID..Spleenb6.meta.data.stage..
ggplot(data1, aes(fill=OTII_status, y=Percentage, x=stage)) + 
    geom_bar(position="stack", stat="identity", width=0.7)+
   xlab("Stade of leukemia")+ylab("Percentage")+ theme(legend.position="bottom")+scale_fill_manual(values = OTIIpalette) +theme_light()+geom_hline(yintercept=c(25,50,75), linetype="dashed", color = "black")+ annotate("text", x = 1:3, y=103, label = c(text)) + labs(title = "Distribution of OTII status in Spleen B6 OTII ")

```
```{r fig.width=10, fig.height=4, warning=TRUE}
DimPlot(Spleenb6, group.by = "OTIIstatus", split.by = "stage", reduction = "umap" ,cols = OTIIpalette, pt.size = 0.5, ncol= 3) + labs(title = "OTII status")
```

# ```{r}
# Idents(SB6.bar) <- "OTIIstatus"
# SB6.OTII <- subset(SB6.bar, idents = c("OTII","OTII - multichains", "non OTII"))
# Idents(SB6.OTII) <- "stage"
# SpleenWTb6 <- subset(SB6.OTII, idents = c("Spleen-WT"))
# df_radarchartWT<-as.data.frame(prop.table(table(SpleenWTb6@meta.data$sig.annotation,SpleenWTb6@meta.data$OTIIstatus),2)*100)
# colnames(df_radarchartWT) <- c( "Annotation","OTIIstatus","Proportion")
# df_radarchartWT <- subset(df_radarchartWT, df_radarchartWT$Annotation %in% c("SP4 mature","DPblast-Myc+","DPsm-Myc+", "SP4-Myc+","Undefined Myc+","Tgd/NK/NKT","NK"))
# 
# SpleenPreTumb6 <- subset(SB6.OTII, idents = c("Spleen-PreTum"))
# df_radarchartPreTum<-as.data.frame(prop.table(table(SpleenPreTumb6@meta.data$sig.annotation,SpleenPreTumb6@meta.data$OTIIstatus),2)*100)
# colnames(df_radarchartPreTum) <- c( "Annotation","OTIIstatus","Proportion")
# df_radarchartPreTum <- subset(df_radarchartPreTum, df_radarchartPreTum$Annotation %in% c("SP4 mature","DPblast-Myc+","DPsm-Myc+", "SP4-Myc+","Undefined Myc+","Tgd/NK/NKT","NK"))
# 
# SpleenTumb6 <- subset(SB6.OTII, idents = c("Spleen-Tum"))
# df_radarchartTum<-as.data.frame(prop.table(table(SpleenTumb6@meta.data$sig.annotation,SpleenTumb6@meta.data$OTIIstatus),2)*100)
# colnames(df_radarchartTum) <- c( "Annotation","OTIIstatus","Proportion")
# df_radarchartTum <- subset(df_radarchartTum, df_radarchartTum$Annotation %in% c("SP4 mature","DPblast-Myc+","DPsm-Myc+", "SP4-Myc+","Undefined Myc+","Tgd/NK/NKT","NK"))
# ```
# 
# ```{r}
# # Charger les données
# df_list <- list(df_radarchartWT, df_radarchartPreTum, df_radarchartTum)
# df_combined <- NULL
#   for (i in seq_along(df_list)) {
#     df_i <- subset(df_list[[i]], OTIIstatus == "OTII - multichains")
#       df_ti <- data.frame(t(df_i[,"Proportion"]))
#       colnames(df_ti) <- df_i$Annotation
#       # Ajouter chaque data frame transformé au data frame combiné
#       df_combined <- rbind(df_combined, setNames(df_ti, colnames(df_ti)))
#       col_names <- c(col_names, paste(colnames(df_ti), labels[i], sep=" - "))
#   }
# df_combined <- rbind(rep(100, 7), rep(0, 7),df_combined)[, 1:7]
# rownames(df_combined) <- c("Max","Min","Spleen-WT","Spleen-PreTum","Spleen-Tum")
# op <- par(mar = c(1, 1, 1, 1))  # Ajuster les marges
# radarchart(df_combined, cglcol="black", cglty=2, cglwd=1, vlcex=1,
#                pcol="#FDE725FF",pty = 20, plty=1:3, 
#                caxislabels=paste(seq(from=0, to=100, by=25), "%"),
#                axislabcol="black", axistype=4,
#                title=paste("OTII - multichains"),
#                plwd=4)
# 
# # Ajouter une légende
# legend("topright", legend=rownames(df_combined)[3:5], lty=linetypes, lwd=2, cex=0.8)
# node_labels <- colnames(df_combined)
# node_positions <- seq(0, 2 * pi, length.out = length(node_labels) + 1)[-1]
# node_r <- rep(60, length(node_labels))  # Distance pour chaque nœud
# text_pos_x <- cos(node_positions) * (node_r + 5)  # Ajuster la position x
# text_pos_y <- sin(node_positions) * (node_r + 5)  # Ajuster la position y
# text_labels <- colnames(df_combined)
# 
# # Tracer les étiquettes
# text(text_pos_x, text_pos_y, labels = text_labels, cex = 0.8)
# 
# par(op)
# ```

```{r fig.width=14, fig.height=6, warning=TRUE}
FeaturePlot(b6.combined.cc, features = c("Trbv12-2","adt_TCR-VB5","adt_TCR-B","Trbv26","Trbv15","Trbv29","Trbv1","Trbv13-2"), reduction = "umap", pt.size = 0.2, ncol = 4, order = T, max.cutoff = 4,keep.scale = "all" )& scale_colour_gradientn(colours = rev(viridis::rocket(10))) &
    NoLegend() + 
    theme_void()+
    theme(plot.title = element_text(hjust = 0.5))
```

```{r fig.width=14, fig.height=8, warning=TRUE}
b6.combined.cc$miceID <- factor(b6.combined.cc$miceID, levels=c("T497","T499","S497+499","500","498","585","578","580","582"))
VlnPlot(b6.combined.cc, features = c("Trbv12-2","Trbv26","Trbv15","Trbv29","Trbv1","Trbv13-2"), ncol = 3,group.by = "miceID", pt.size = 0) & geom_boxplot(width=0.1, fill="white")
```


```{r}
Idents(b6.combined.cc) <-"RNA_snn_res.1"
FeaturePlot(b6.combined.cc, features = c("adt_TCR-VB5"),order = T, reduction = "umap", pt.size = 0.2, ncol = 2 ) & scale_colour_gradientn(colours = rev(viridis::rocket(10)))
VlnPlot(b6.combined.cc, features = c("adt_TCR-VB5") ,group.by = "RNA_snn_res.1", pt.size = 0, sort =T , idents = c("14","10")) & geom_boxplot(width=0.1, fill="white")
```

# TCR BC
```{r}
Idents(bc.combined.cc) <- "OTIIstatus"
OTIIbc <- subset(bc.combined.cc, idents = c("OTII","OTII - multichains","non OTII"))# Remove 'unknown' and 'OTII- incomplete" because not informative
Idents(OTIIbc) <- "annotation"
Thymusbc <- subset(OTIIbc, idents = c("Thymus-WT","Thymus-PreTum late","Thymus-Tum"))
data1 <- as.data.frame(prop.table(table(Thymusbc@meta.data$OTIIstatus,Thymusbc@meta.data$annotation),2)*100)
#data1 <-na.omit(data1)
data1 <- data1[!(data1$Freq=="NaN"), ]
#order leukestage level
data1$Var2 <- factor(data1$Var2, levels = c("Thymus-WT","Thymus-PreTum late","Thymus-Tum"))
cellnumber <- data.frame(colSums(table(Thymusbc@meta.data$MULTI_ID,Thymusbc@meta.data$annotation)))
cellnumber <- subset(cellnumber,rownames(cellnumber) %in% c("Thymus-WT","Thymus-PreTum late","Thymus-Tum"))
cellnumber$annotation <- rownames(cellnumber)
row_order <-c("Thymus-WT","Thymus-PreTum late","Thymus-Tum")
cellnumber <- cellnumber[row_order,]
annotation <- data1$Var2
OTII_status <- data1$Var1
OTII_status <- factor(OTII_status, levels = rev(c("OTII","OTII - multichains", "non OTII"))) 
Percentage <- data1$Freq
text <- cellnumber$colSums.table.Thymusbc.meta.data.MULTI_ID..Thymusbc.meta.data.annotation..
ggplot(data1, aes(fill=OTII_status, y=Percentage, x=annotation)) + 
    geom_bar(position="stack", stat="identity", width=0.7)+
   xlab("Stade of leukemia")+ylab("Percentage")+ theme(legend.position="bottom")+scale_fill_manual(values = OTIIpalette) +theme_light()+geom_hline(yintercept=c(25,50,75), linetype="dashed", color = "black")+ annotate("text", x = 1:3, y=103, label = c(text)) + labs(title = "Distribution of OTII status in Thymus BC OTII ")

```
```{r fig.width=12, fig.height=4, warning=TRUE}
DimPlot(Thymusbc, group.by = "OTIIstatus", split.by = "annotation", reduction = "umap" ,cols = OTIIpalette, pt.size = 0.5, ncol= 3) + labs(title = "OTII status")
```

```{r}
Idents(OTIIbc) <- "annotation"
Spleenbc <- subset(OTIIbc, idents = c("Spleen-WT","Spleen-PreTum early","Spleen-PreTum late","Spleen-Tum"))
data1 <- as.data.frame(prop.table(table(Spleenbc@meta.data$OTIIstatus,Spleenbc@meta.data$annotation),2)*100)
#data1 <-na.omit(data1)
data1 <- data1[!(data1$Freq=="NaN"), ]
#order leukestage level
data1$Var2 <- factor(data1$Var2, levels = c("Spleen-WT","Spleen-PreTum early","Spleen-PreTum late","Spleen-Tum"))
cellnumber <- data.frame(colSums(table(Spleenbc@meta.data$MULTI_ID,Spleenbc@meta.data$annotation)))
cellnumber <- subset(cellnumber,rownames(cellnumber) %in% c("Spleen-WT","Spleen-PreTum early","Spleen-PreTum late","Spleen-Tum"))
cellnumber$annotation <- rownames(cellnumber)
row_order <-c("Spleen-WT","Spleen-PreTum early","Spleen-PreTum late","Spleen-Tum")
cellnumber <- cellnumber[row_order,]
annotation <- data1$Var2
OTII_status <- data1$Var1
OTII_status <- factor(OTII_status, levels = rev(c("OTII","OTII - multichains", "non OTII"))) 
Percentage <- data1$Freq
text <- cellnumber$colSums.table.Spleenbc.meta.data.MULTI_ID..Spleenbc.meta.data.annotation..
ggplot(data1, aes(fill=OTII_status, y=Percentage, x=annotation)) + 
    geom_bar(position="stack", stat="identity", width=0.7)+
   xlab("Stade of leukemia")+ylab("Percentage")+ theme(legend.position="bottom")+scale_fill_manual(values = OTIIpalette) +theme_light()+geom_hline(yintercept=c(25,50,75), linetype="dashed", color = "black")+ annotate("text", x = 1:4, y=103, label = c(text)) + labs(title = "Distribution of OTII status in Spleen BC OTII ")

```

```{r fig.width=8, fig.height=4, warning=TRUE}
DimPlot(Spleenbc, group.by = "OTIIstatus", split.by = "annotation", reduction = "umap" ,cols = OTIIpalette, pt.size = 0.5, ncol= 2) + labs(title = "OTII status")
```

```{r}
# Idents(bc.combined.cc) <- "OTIIstatus"
# OTIIbc <- subset(bc.combined.cc, idents = c("OTII","OTII - multichains","non OTII"))
# Idents(OTIIbc) <- "annotation"
# Thymusbc <- subset(OTIIbc, idents = c("Thymus-WT","Thymus-PreTum late","Thymus-Tum"))
# ThymusWTbc <- subset(Thymusbc, idents = c("Thymus-WT"))
# df_radarchartWT<-as.data.frame(prop.table(table(ThymusWTbc@meta.data$sig.annotation,ThymusWTbc@meta.data$OTIIstatus),2)*100)
# colnames(df_radarchartWT) <- c( "Annotation","OTIIstatus","Proportion")
# 
# ThymusPreTumbc <- subset(Thymusbc, idents = c("Thymus-PreTum late"))
# df_radarchartPreTum<-as.data.frame(prop.table(table(ThymusPreTumbc@meta.data$sig.annotation,ThymusPreTumbc@meta.data$OTIIstatus),2)*100)
# colnames(df_radarchartPreTum) <- c( "Annotation","OTIIstatus","Proportion")
# 
# ThymusTumbc <- subset(Thymusbc, idents = c("Thymus-Tum"))
# df_radarchartTum<-as.data.frame(prop.table(table(ThymusTumbc@meta.data$sig.annotation,ThymusTumbc@meta.data$OTIIstatus),2)*100)
# colnames(df_radarchartTum) <- c( "Annotation","OTIIstatus","Proportion")
 ```

```{r}
# # Charger les données
# df_list <- list(df_radarchartWT, df_radarchartPreTum, df_radarchartTum)
# df_combined <- NULL
#   for (i in seq_along(df_list)) {
#     df_i <- subset(df_list[[i]], OTIIstatus == "OTII")
#       df_ti <- data.frame(t(df_i[,"Proportion"]))
#       colnames(df_ti) <- df_i$Annotation
#       # Ajouter chaque data frame transformé au data frame combiné
#       df_combined <- rbind(df_combined, setNames(df_ti, colnames(df_ti)))
#       col_names <- c(col_names, paste(colnames(df_ti), labels[i], sep=" - "))
#   }
# df_combined <- rbind(rep(60, 21), rep(0, 21),df_combined)[, 1:21]
# rownames(df_combined) <- c("Max","Min","Thymus-WT","Thymus-PreTum","Thymus-Tum")
# op <- par(mar = c(1, 1, 1, 1))  # Ajuster les marges
# radarchart(df_combined, cglcol="black", cglty=2, cglwd=1, vlcex=1,
#                pcol="#7AD151FF",pty = 20, plty=1:3, 
#                caxislabels=paste(seq(from=0, to=60, by=15), "%"),
#                axislabcol="black", axistype=4,
#                title=paste("OTII"),
#                plwd=4)
# 
# # Ajouter une légende
# #legend("topright", legend=rownames(df_combined)[3:5], lty=linetypes, lwd=2, cex=0.8)
# node_labels <- colnames(df_combined)
# node_positions <- seq(0, 2 * pi, length.out = length(node_labels) + 1)[-1]
# node_r <- rep(60, length(node_labels))  # Distance pour chaque nœud
# text_pos_x <- cos(node_positions) * (node_r + 5)  # Ajuster la position x
# text_pos_y <- sin(node_positions) * (node_r + 5)  # Ajuster la position y
# text_labels <- colnames(df_combined)
# 
# # Tracer les étiquettes
# text(text_pos_x, text_pos_y, labels = text_labels, cex = 0.8)
# 
# par(op)
```


```{r}
# Idents(maturebar) <- "OTIIstatus"
# SBC.OTII <- subset(maturebar, idents = c("OTII","OTII - multichains", "non OTII"))
# Idents(SBC.OTII) <- "annotation"
# SpleenWTbc <- subset(SBC.OTII, idents = c("Spleen-WT"))
# df_radarchartWT<-as.data.frame(prop.table(table(SpleenWTbc@meta.data$sig.annotation,SpleenWTbc@meta.data$OTIIstatus),2)*100)
# colnames(df_radarchartWT) <- c( "Annotation","OTIIstatus","Proportion")
# df_radarchartWT <- subset(df_radarchartWT, df_radarchartWT$Annotation %in% c("SP8","SP4","SP4CD69+","Tgd early","Tgd mature","MAIT","NK","Undefined Myc+", "Dying cells"))
# 
# SpleenPreTumbc <- subset(SBC.OTII, idents = c("Spleen-PreTum late"))
# df_radarchartPreTum<-as.data.frame(prop.table(table(SpleenPreTumbc@meta.data$sig.annotation,SpleenPreTumbc@meta.data$OTIIstatus),2)*100)
# colnames(df_radarchartPreTum) <- c( "Annotation","OTIIstatus","Proportion")
# df_radarchartPreTum <- subset(df_radarchartPreTum, df_radarchartPreTum$Annotation %in% c("SP8","SP4","SP4CD69+","Tgd early","Tgd mature","MAIT","NK","Undefined Myc+", "Dying cells"))
# 
# SpleenTumbc <- subset(SBC.OTII, idents = c("Spleen-Tum"))
# df_radarchartTum<-as.data.frame(prop.table(table(SpleenTumbc@meta.data$sig.annotation,SpleenTumbc@meta.data$OTIIstatus),2)*100)
# colnames(df_radarchartTum) <- c( "Annotation","OTIIstatus","Proportion")
# df_radarchartTum <- subset(df_radarchartTum, df_radarchartTum$Annotation %in% c("SP8","SP4","SP4CD69+","Tgd early","Tgd mature","MAIT","NK","Undefined Myc+", "Dying cells"))
```

```{r}
# # Charger les données
# df_list <- list(df_radarchartWT, df_radarchartPreTum, df_radarchartTum)
# df_combined <- NULL
#   for (i in seq_along(df_list)) {
#     df_i <- subset(df_list[[i]], OTIIstatus == "non OTII")
#       df_ti <- data.frame(t(df_i[,"Proportion"]))
#       colnames(df_ti) <- df_i$Annotation
#       # Ajouter chaque data frame transformé au data frame combiné
#       df_combined <- rbind(df_combined, setNames(df_ti, colnames(df_ti)))
#       col_names <- c(col_names, paste(colnames(df_ti), labels[i], sep=" - "))
#   }
# df_combined <- rbind(rep(100, 9), rep(0, 9),df_combined)[, 1:9]
# rownames(df_combined) <- c("Max","Min","Spleen-WT","Spleen-PreTum","Spleen-Tum")
# op <- par(mar = c(1, 1, 1, 1))  # Ajuster les marges
# radarchart(df_combined, cglcol="black", cglty=2, cglwd=1, vlcex=1,
#                pcol="#DE4968FF",pty = 20, plty=1:3, 
#                caxislabels=paste(seq(from=0, to=100, by=25), "%"),
#                axislabcol="black", axistype=4,
#                title=paste("non OTII"),
#                plwd=4)
# 
# # Ajouter une légende
# #legend("topright", legend=rownames(df_combined)[3:5], lty=linetypes, lwd=2, cex=0.8)
# node_labels <- colnames(df_combined)
# node_positions <- seq(0, 2 * pi, length.out = length(node_labels) + 1)[-1]
# node_r <- rep(60, length(node_labels))  # Distance pour chaque nœud
# text_pos_x <- cos(node_positions) * (node_r + 5)  # Ajuster la position x
# text_pos_y <- sin(node_positions) * (node_r + 5)  # Ajuster la position y
# text_labels <- colnames(df_combined)
# 
# # Tracer les étiquettes
# text(text_pos_x, text_pos_y, labels = text_labels, cex = 0.8)
# 
# par(op)
```

```{r, fig.width=12, fig.height=8, warning=FALSE}
FeaturePlot(bc.combined.cc, features = c("adt_TCR-VA2","adt_TCR-VB5"),order = T, reduction = "wnn.umap", pt.size = 0.2, ncol = 2 ) & scale_colour_gradientn(colours = rev(viridis::rocket(10)))
Idents (bc.combined.cc) <- "annotation"
DimPlot(bc.combined.cc, group.by = "MULTI_ID",split.by="miceID",order = rev(c("51","83","47","85","84","39","52","70","80","53","69")) , reduction = "wnn.umap", pt.size = 0.5, label = TRUE, repel = TRUE, label.size = 2.5, ncol = 4)
```


```{r}
# data1 <- as.data.frame(prop.table(table(OTIIbc@meta.data$OTIIstatus,OTIIbc@meta.data$annotation),2)*100)
# data1 <-na.omit(data1)
# #data1 <- data1[!is.na(data1$Freq), ]
# 
# #order leukestage level
# data1$Var2 <- factor(data1$Var2, levels = c("Thymus-WT","Thymus-PreTum early","Thymus-PreTum late","Thymus-Tum"))
# cellnumber <- data.frame(colSums(table(OTIIbc@meta.data$MULTI_ID,OTIIbc@meta.data$annotation)))
# 
# cellnumber$stage <- rownames(cellnumber)
# row_order <-c("Thymus-WT","Thymus-PreTum early","Thymus-PreTum late","Thymus-Tum")
# cellnumber <- cellnumber[row_order,]
# annotaion <- data1$Var2
# OTII_status <- data1$Var1
# OTII_status <- factor(OTII_status, levels = rev(c("OTII","OTII - multichains", "non OTII"))) 
# Percentage <- data1$Freq
# text <- cellnumber$colSums.table.OTIIbc.meta.data.MULTI_ID..OTIIbc.meta.data.annotation..
# ggplot(data1, aes(fill=OTII_status, y=Percentage, x=annotation)) + 
#     geom_bar(position="stack", stat="identity", width=0.7)+
#    xlab("Stade of leukemia")+ylab("Percentage")+ theme(legend.position="bottom")+scale_fill_manual(values = OTIIpalette) +theme_light()+geom_hline(yintercept=c(25,50,75), linetype="dashed", color = "black")+ annotate("text", x = 1:4, y=103, label = c(text)) + labs(title = "Distribution of OTII status in Thymus BC OTII ")
# DimPlot(OTIIbc, group.by = "OTIIstatus", split.by = "stage", reduction = "umap" ,cols = OTIIpalette, pt.size = 0.5, ncol= 3) + labs(title = "OTII status")
```


```{r, warning=FALSE}
Idents(bc.combined.cc) <- "sig.annotation"
VlnPlot(bc.combined.cc, features = c("Trbv12-2"), ncol = 1,group.by = "sig.annotation",idents = c("DPsm","DPsmCD25+","DPsmCCR7+","DP69","SP8","SP4","SP4CD69+","DPsm Myc+","DP69 Myc+","Undefined Myc+") , pt.size = 0, y.max = 4, cols = c(palette1,palette2,palette3,palette4) )& theme(plot.title = element_text(hjust = 0.5,face = "bold")) & geom_boxplot(width=0.1, fill="white")
```

```{r, warning=FALSE}
Idents(bc.combined.cc) <- "miceID"
bc.combined.cc$miceID <- factor(bc.combined.cc$miceID, levels=c("51","83","39","47","52","84","85","70","80","53","69"))
VlnPlot(bc.combined.cc, features = c("adt_TCR-VB5"), ncol = 1,group.by = "miceID", pt.size = 0, y.max = 3, cols = c(rep("#9AC4F8",2),rep("#fac341",6),rep("#BB0A21",3)))& theme(plot.title = element_text(hjust = 0.5,face = "bold")) & geom_boxplot(width=0.1, fill="white")
Idents(bc.combined.cc) <- "miceID"
VlnPlot(bc.combined.cc, features = c("adt_TCR-B"), ncol = 1,group.by = "miceID", pt.size = 0,y.max = 3, cols = c(rep("#9AC4F8",2),rep("#fac341",6),rep("#BB0A21",3)))& theme(plot.title = element_text(hjust = 0.5,face = "bold")) & geom_boxplot(width=0.1, fill="white")
```

```{r, fig.width=12, fig.height=8, warning=FALSE}
Idents(bc.combined.cc) <- "miceID"
VlnPlot(bc.combined.cc, features = c("Trbv17","Trbv16","Trbv31","Trbv2","Trbv20","Trbv15","Trbv13-1"),group.by = "miceID", pt.size = 0, ncol = 3)+ theme(legend.position = "none")
```

```{r, fig.width=12, fig.height=4, warning=FALSE}
VlnPlot(bc.combined.cc, features = c("Trbv16","Trbv2","Trbv15","Trbv13-1"),group.by = "miceID", pt.size = 0,idents = c("39","47","52","84","85","70"), ncol = 3, cols= rep("#fac341",6))+ theme(legend.position = "none")
VlnPlot(bc.combined.cc, features = c("Trbv17","Trbv31","Trbv20"), group.by = "miceID", pt.size = 0, ncol = 3, idents = c("80","53","69"), cols = rep("#BB0A21",3) )+ theme(legend.position = "none")
```



# DEG analysis 
## B6
### TumvsPreTum

```{r}
degb6_TumvsPreTum.T <- read.csv(file = "/mnt/NASBIOINFO_CP/LALT/BIOINFO/PTEN/03_Analysis/Analysis2024/outputs/B6/degb6_TumvsPreTum.T.csv")

upreg_TumvsPreTumb6 <-extract_upDEGs(degb6_TumvsPreTum.T,p_val_adj_cutoff = 1e-20 ,pct_cutoff = 0.3 , avglog2fc_cutoff = 1,d.pct_cutoff = 0)
downreg_TumvsPreTumb6 <- extract_downDEGs(degb6_TumvsPreTum.T,p_val_adj_cutoff = 1e-20 ,pct_cutoff = 0.3 , avglog2fc_cutoff = -1,d.pct_cutoff = 0)
```


```{r}
ordered_upreg_TumvsPreTumb6 <- upreg_TumvsPreTumb6 %>%
  arrange(p_val_adj) %>%
  dplyr::select(gene, Name)

ordered_downreg_TumvsPreTumb6 <- downreg_TumvsPreTumb6 %>%
  arrange(p_val_adj) %>%
  top_n(n=50, wt = -p_val_adj) %>%
  dplyr::select(gene)
```


# Volcano plot
```{r, fig.width=8, fig.height=6, warning=FALSE}
markers <- degb6_TumvsPreTum.T
# Remplacer les valeurs de p_val_adj égales à 0 par une petite valeur positive
markers$p_val_adj[markers$p_val_adj == 0] <- 1e-315

markers$diffexpressed <- "NO"
markers$diffexpressed[markers$avg_log2FC > 1 & markers$p_val_adj < 1e-20] <- "UP"
markers$diffexpressed[markers$avg_log2FC < -1 & markers$p_val_adj <1e-20] <- "DOWN"
# Sélectionner les 20 gènes les plus significatifs
top_genes <- markers %>% filter((p_val_adj < 1e-200 | abs(avg_log2FC) > 3) & !(markers$diffexpressed == "NO"))
markers$delabel <- ifelse(markers$gene %in% top_genes$gene, markers$gene, NA)



volcano_plot <- ggplot(data = markers, aes(x = avg_log2FC, y = -log10(p_val_adj), col = diffexpressed, label = delabel)) +
  geom_vline(xintercept = c(-1,1 ), col = "gray", linetype = 'dashed') +
  geom_hline(yintercept = -log10(1e-20), col = "gray", linetype = 'dashed') +
  geom_point(size = 2) +
  scale_color_manual(values = c("#00AFBB", "grey", "#bb0c00"), # to set the colours of our variable
                     labels = c("Downregulated", "Not significant", "Upregulated")) + # to set the labels in case we want to overwrite the categories from the dataframe (UP, DOWN, NO)
  coord_cartesian(ylim = c(0, 350), xlim = c(-6, 6)) + # since some genes can have minuslog10padj of inf, we set these limits
  labs(color = 'DEG', #legend_title,
       x = expression("avg_log"[2]*"FC Tum vs PreTum"), y = expression("-log"[10]*"(p_val_adj)")) +
  scale_x_continuous(breaks = seq(-6, 6, 2)) + # to customise the breaks in the x axis
  #ggtitle('Thf-like cells in severe COVID vs healthy patients') + # Plot title
  geom_text_repel(max.overlaps = Inf) # To show all labels 
# Afficher le plot
print(volcano_plot)
```
The following plots show the expression of top DE genes based on avg_logFC

#Functional enrichment of B6 DEG

```{r}
backgroundb6 <- b6.combined.cc@assays$RNA@features
backgroundrowb6 <- rownames(backgroundb6)
CPenrich <- enrichGO(downreg_TumvsPreTumb6$gene, OrgDb = 'org.Mm.eg.db', ont="BP",keyType = "SYMBOL",universe = backgroundrowb6) # org.Mm.eg.db genome mouse
barplot(CPenrich,x="Count", showCategory=15,decrasing = FALSE)
# mutate(CPenrich, qscore = -log(p.adjust, base=10)) %>% 
#     barplot(x="qscore",showCategory=15)
# 
# dotplot(CPenrich, showCategory=20,color = "p.adjust",x="Count")+ ggtitle("GO Enrichment") #+ coord_flip()+theme(axis.text.x = element_text(angle = 90, hjust = 1))
#cnetplot(CPenrich, node_label="all", circular = F, color.params = list(foldChange = NULL , edge = FALSE , category = "#E5C494", gene = "#B3B3B3"))
# Convert gene symbol to gene ID
entrez_ids <- bitr(downreg_TumvsPreTumb6$gene, fromType = "SYMBOL", toType = "ENTREZID", OrgDb = org.Mm.eg.db)
kegg <- enrichKEGG(gene = entrez_ids$ENTREZID,
                 organism = 'mmu',
                 pAdjustMethod = "BH",
                 qvalueCutoff = 0.05,
                 minGSSize = 10)
# dotplot(kegg) + ggtitle("KEGG Pathway Enrichment")
# CPenrich_ema <- pairwise_termsim(CPenrich)
# treeplot(CPenrich_ema, hclust_method = "average")
# emapplot(CPenrich_ema)

```

```{r, fig.width=8, fig.height=8, warning=FALSE}
# Filtrer les résultats KEGG pour ne conserver que ceux significatifs (p.adjust < 0.05 par exemple)
significant_kegg <- kegg@result %>%
  filter(Count >= 11 & !(category == "Human Diseases"))
# Nettoyer les descriptions des voies KEGG
significant_kegg$Description <- gsub(" - Mus musculus \\(house mouse\\)", "", significant_kegg$Description)

# Préparer les données pour le diagramme circos
significant_kegg$geneID <- strsplit(as.character(significant_kegg$geneID), "/")

# Convertir les identifiants Entrez en symboles de gènes pour le diagramme
gene_conversion <- bitr(unlist(significant_kegg$geneID), fromType="ENTREZID", toType="SYMBOL", OrgDb=org.Mm.eg.db)
conversion_table <- setNames(gene_conversion$SYMBOL, gene_conversion$ENTREZID)

# Remplacer les identifiants dans significant_kegg par les symboles de gènes
significant_kegg$geneID <- lapply(significant_kegg$geneID, function(ids) {
  sapply(ids, function(id) conversion_table[as.character(id)])
})

# Créer une table des relations gène-voie avec les symboles de gènes
df <- data.frame(
  pathway = rep(significant_kegg$Description, sapply(significant_kegg$geneID, length)),
  gene = unlist(significant_kegg$geneID)
)

# Réduire l'écart entre les secteurs
circos.par(gap.degree = 1)  # Ajustez cette valeur si nécessaire

# Créer le diagramme circos avec circlize
circos.clear()
chordDiagram(df, annotationTrack = "grid", preAllocateTracks = 1)

# Ajouter les annotations
circos.track(track.index = 1, panel.fun = function(x, y) {
    circos.text(CELL_META$xcenter, CELL_META$ylim[1], CELL_META$sector.index, facing = "clockwise", niceFacing = TRUE, adj = c(0, 0.5))
}, bg.border = NA)
```


#DEG BC
```{r}
cluster21vs9.T <- read.csv("/mnt/NASBIOINFO_CP/LALT/BIOINFO/PTEN/03_Analysis/Analysis2024/outputs/BC/DEG_cluster21vs9.T.csv", h= T, sep = ",")
cluster7vs24.T <- read.csv("/mnt/NASBIOINFO_CP/LALT/BIOINFO/PTEN/03_Analysis/Analysis2024/outputs/BC/DEG_cluster7vs24.T.csv", h= T, sep = ",")
cluster3vs1.T <- read.csv("/mnt/NASBIOINFO_CP/LALT/BIOINFO/PTEN/03_Analysis/Analysis2024/outputs/BC/DEG_cluster3vs1.T.csv", h= T, sep = ",") 
degbc.list <- list(cluster3vs1.T,cluster7vs24.T,cluster21vs9.T)
```

```{r}
Idents(bc.combined.cc) <- "miceID"
tumbc.mice <- subset(bc.combined.cc,idents = c("80","53","69"))
Idents(tumbc.mice) <- "wsnn_res.2"
tumbc.mice <- subset(tumbc.mice,idents = c("1","3","7","24","21","9"))

tum.cols <- c("#9063CDFF","#702F8AFF","#7EA7D3FF","#13317DFF","#ED968CFF","#FF6720FF")
names(tum.cols) <- c("Thymus-80","Spleen-80","Thymus-53","Spleen-53","Thymus-69","Spleen-69")
tum.clusters.cols <- c("#9063CDFF","#702F8AFF","#7EA7D3FF","#13317DFF","#ED968CFF","#FF6720FF")
names(tum.clusters.cols) <- c("1","3","7","24","21","9")

DimPlot(tumbc.mice, group.by = "MULTI_ID",reduction = "wnn.umap",cols=tum.cols, pt.size = 1)&
    NoLegend() + 
    theme_void()+
    theme(plot.title = element_text(hjust = 0.5))

DimPlot(tumbc.mice, group.by = "wsnn_res.2",reduction = "wnn.umap",cols=tum.clusters.cols, pt.size = 1)&
    NoLegend() + 
    theme_void()+
    theme(plot.title = element_text(hjust = 0.5))

```


```{r, fig.width=5, fig.height=5, warning=FALSE}
# deg80_downreg <- extract_DEGs(cluster1vs3,p_val_adj_cutoff = 1e-10 ,pct_cutoff = 0.05 ,d.pct_cutoff = 0, avglog2fc_cutoff = 1 , direction = "downregulated" )
# deg69_downreg <- extract_DEGs(cluster9vs21,p_val_adj_cutoff = 1e-10 ,pct_cutoff = 0.1 ,d.pct_cutoff = 0, avglog2fc_cutoff = 1 , direction = "downregulated" )
# deg53_downreg <- extract_DEGs(cluster24vs7,p_val_adj_cutoff = 1e-10 ,pct_cutoff = 0.1 ,d.pct_cutoff = 0, avglog2fc_cutoff = 1 , direction = "downregulated" )
# deg.down.list <- list(deg80_downreg$gene, deg69_downreg$gene, deg53_downreg$gene)
# 
# deg80_downreg <- extract_DEGs(cluster1vs3,p_val_adj_cutoff = 0.05 ,pct_cutoff = 0 ,d.pct_cutoff = 0.15, avglog2fc_cutoff = 0.2 , direction = "upregulated" )
# deg69_downreg <- extract_DEGs(cluster9vs21,p_val_adj_cutoff = 0.05 ,pct_cutoff = 0 ,d.pct_cutoff = 0.15, avglog2fc_cutoff = 0.2 , direction = "upregulated" )
# deg53_downreg <- extract_DEGs(cluster24vs7,p_val_adj_cutoff = 0.05 ,pct_cutoff = 0 ,d.pct_cutoff = 0.15, avglog2fc_cutoff = 0.2 , direction = "upregulated" )
# deg.down.list <- list(deg80_downreg$gene, deg69_downreg$gene, deg53_downreg$gene)

deg80_upreg <- extract_upDEGs(cluster3vs1.T,p_val_adj_cutoff = 0.05 ,pct_cutoff = 0.05 ,d.pct_cutoff = 0, avglog2fc_cutoff = 1 )
deg69_upreg <- extract_upDEGs(cluster21vs9.T,p_val_adj_cutoff = 0.05 ,pct_cutoff = 0.05 ,d.pct_cutoff = 0, avglog2fc_cutoff = 1 )
deg53_upreg <- extract_upDEGs(cluster7vs24.T,p_val_adj_cutoff = 0.05 ,pct_cutoff = 0.05 ,d.pct_cutoff = 0, avglog2fc_cutoff = 1 )
deg.up.list <- list(deg80_upreg$gene, deg69_upreg$gene, deg53_upreg$gene)

deg80_downreg <- extract_downDEGs(cluster3vs1.T,p_val_adj_cutoff = 0.05 ,pct_cutoff = 0.05 , avglog2fc_cutoff = -1,d.pct_cutoff = 0 )
deg69_downreg <- extract_downDEGs(cluster21vs9.T,p_val_adj_cutoff = 0.05 ,pct_cutoff = 0.05 , avglog2fc_cutoff = -1,d.pct_cutoff = 0)
deg53_downreg <- extract_downDEGs(cluster7vs24.T,p_val_adj_cutoff = 0.05 ,pct_cutoff = 0.05 , avglog2fc_cutoff = -1,d.pct_cutoff = 0)
deg.down.list <- list(deg80_downreg$gene, deg69_downreg$gene, deg53_downreg$gene)
display_venn <- function(x, names,cols){
  library(VennDiagram)
  grid.newpage()
  venn_object <- venn.diagram(x, filename = NULL,
        category.names = names,
        # Circles
        lwd = 2,
        lty = 'blank',
        fill = cols ,
        # Numbers
        cex = .9,
        fontface = "italic",
        # Set names
        cat.cex = 1,
        cat.fontface = "bold",
        cat.default.pos = "outer",
        #Title
        main.title = name)
  grid.draw(venn_object)
}
display_venn(deg.up.list, names =c("DEG 80","DEG 69","DEG 53"), cols =c("#121510FF","#6D8325FF","#D6CFB7FF" ))
display_venn(deg.down.list, name = c("DEG 80","DEG 69","DEG 53"), cols =c("#A63228FF","#C7662AFF","#EBA42BFF"))

# DEG commons between 3 mice 
upreg.Tum <-  Reduce(intersect, list(deg80_upreg$gene, deg69_upreg$gene, deg53_upreg$gene)) 
downreg.Tum <- Reduce(intersect, list(deg80_downreg$gene, deg69_downreg$gene, deg53_downreg$gene)) 


```


```{r}
# Scoring of down- and up- geneset
exprMatrix <- bc.combined.cc@assays$RNA$data
geneRankings <- AUCell_buildRankings(exprMatrix, plotStats=TRUE)
aucellScores <- AUCell_calcAUC(downreg.Tum, geneRankings)
aucellScores2 <- AUCell_calcAUC(upreg.Tum, geneRankings)
aucScoresMatrix <- aucellScores@assays@data$AUC
aucScoresMatrix2 <- aucellScores2@assays@data$AUC
# Create a new assay object with the matrix
AUC <- CreateAssayObject(data = aucScoresMatrix)
AUC2 <- CreateAssayObject(data = aucScoresMatrix2)
bc.combined.cc[['DEGdown']] <- AUC
bc.combined.cc[['DEGup']] <- AUC2
VlnPlot(bc.combined.cc, features = "degdown_geneSet", group.by = "annotation", pt.size = 0)& geom_boxplot(width=0.1, fill="white")
VlnPlot(bc.combined.cc, features = "degup_geneSet", group.by = "annotation", pt.size = 0)& geom_boxplot(width=0.1, fill="white")
```



```{r}
ordered.upreg.Tum <- cluster3vs1.T %>%
  filter(gene %in% upreg.Tum) %>%
  arrange(desc(avg_log2FC)) %>%
  dplyr::select(gene, Name)

ordered.downreg.Tum <- cluster3vs1.T %>%
  filter(gene %in% downreg.Tum) %>%
  arrange(desc(avg_log2FC)) %>%
  dplyr::select(gene, Name)
```

```{r}
backgroundbc <- bc.combined.cc@assays$RNA@features
backgroundrowbc <- rownames(backgroundbc)
CPenrich <- enrichGO(gene= downreg.Tum, OrgDb = 'org.Mm.eg.db', ont="BP",keyType = "SYMBOL",universe = backgroundrowbc) # org.Mm.eg.db genome mouse
dotplot(CPenrich, showCategory=20,color = "p.adjust",x="Count")+ ggtitle("GO Enrichment") #+ coord_flip()+theme(axis.text.x = element_text(angle = 90, hjust = 1))

# 
# terms <- CPenrich@result$Description
# p.adjust_values <- CPenrich@result$p.adjust
# names(p.adjust_values) <- terms
# cnetplot(CPenrich, node_label="all", circular = TRUE,colorEdge=T, color.params = list(foldChange = NULL , edge = FALSE , category = "#E5C494", gene = "#B3B3B3"),hilight.params = list(category = NULL, alpha_hilight = 1, alpha_no_hilight = 0.3))

# Convert gene symbol to gene ID
entrez_ids <- bitr(downreg.Tum, fromType = "SYMBOL", toType = "ENTREZID", OrgDb = org.Mm.eg.db)
kegg <- enrichKEGG(gene = entrez_ids$ENTREZID,
                 organism = 'mmu',
                 pAdjustMethod = "BH",
                 qvalueCutoff = 0.05,
                 minGSSize = 10)
# cnetplot(kegg, node_label="all", circular = TRUE,colorEdge=T, color.params = list(foldChange = NULL , edge = FALSE , category = "#E5C494", gene = "#B3B3B3"))
dotplot(kegg) + ggtitle("KEGG Pathway Enrichment")
```

```{r}
# Intersect of 3 DEGs datasets in BC
common.genes.degbc <- Reduce(intersect, list(cluster21vs9.T$gene,cluster3vs1.T$gene,cluster7vs24.T$gene))
trim_names <- c("cluster21vs9.T","cluster3vs1.T","cluster7vs24.T")
## extract logFC for common genes for each files and merge
for (i in 1:length(degbc.list)) {
  d  <- as.data.frame(degbc.list[[i]])  #d is temp variable
  ##extracting specified coulmns from the files ## these col names should be same in all the files
  names(d) <- c("p_val", "avg_log2FC", "pct.1", "pct.2", "p_val_adj", "gene", "Name", "abs")
  d <- d[!duplicated(d[ , c("gene")]),]
  d <- d[d$gene %in% common.genes.degbc,c("gene","avg_log2FC")]
  names(d) <- c("gene",paste(trim_names[[i]],"avg_log2FC", sep = '_'))
  
  #assign(trim_names[[i]], d)
  assign(paste('d',i,sep = ''), d)
}

Tum_degbc <- Reduce(function(x, y) merge(x, y,by='gene', all=TRUE), list(d1,d2,d3))
#write.table(Tum_degbc,"/mnt/NASBIOINFO_CP/LALT/BIOINFO/PTEN/03_Analysis/Analysis2024/outputs/BC/Tum_degbc.csv", sep = ',', row.names = F)
```

```{r}
DoHeatmap(tumbc.mice, features = upreg.Tum ,colnames(subset(tumbc.mice, idents = levels(c("1","9","24","3","21","7")))),group.colors = tum.clusters.cols, size = 3.5, hjust = 1, raster = FALSE)
```


```{r}
# # Select markers for plotting
# markers.use <- cluster21vs9.T %>%
#     filter(gene %in% c(upreg.Tum,downreg.Tum)) %>%
#     arrange(avg_log2FC)
# markers.use <- mutate(markers.use,
#                         gene = factor(gene, levels = unique(gene)),
#                         logpval = -log10(p_val_adj))
#   # Plot
#   ggplot(markers.use, aes(x = gene, y = avg_log2FC, fill = logpval)) +
#     geom_bar(stat = "identity", width = .6) +
#     ylim(c(-8,5)) +
#     labs(title = "DEG - cluster 21 vs. cluster 9", y = "Log2FC", x = "Genes differentially expressed") +
#     theme_tufte() +  # Assuming ggfortify is installed
#     theme(plot.title = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),axis.text.y = element_text(color = "black"), axis.ticks = element_blank()) +
#     scale_fill_gradient2(low = 'red', mid = 'orange', high = 'blue',midpoint = 30) +
#     coord_flip()  # Flip axes for better visualization
```

# DPsm BC DE genes :
```{r}
degbc_DPsmPtendelvsphysio <- read.csv("/mnt/NASBIOINFO_CP/LALT/BIOINFO/PTEN/03_Analysis/Analysis2024/outputs/BC/DEG_DPsm_Ptendelvsphysio.csv", h= T, sep = ",") 

```

# DP69 BC DE genes : 

```{r}
degbc_DP69Ptendelvsphysio <- read.csv("/mnt/NASBIOINFO_CP/LALT/BIOINFO/PTEN/03_Analysis/Analysis2024/outputs/BC/DEG_DP69_Ptendelvsphysio.csv", h= T, sep = ",") 
degbc_DP69Ptendelvsphysio_up <- degbc_DP69Ptendelvsphysio %>%filter(avg_log2FC>=0.5  & p_val_adj <= 1e-2 & (pct.1> 0.3 | pct.2 >0.3))
degbc_DP69Ptendelvsphysio_down <- degbc_DP69Ptendelvsphysio %>%filter(avg_log2FC<=-0.5 & p_val_adj <= 1e-2 & (pct.1> 0.3 | pct.2 >0.3))
```

# Satb1 
```{r}
FeaturePlot(b6.combined.cc, features = "Satb1" , reduction = "umap", pt.size = 0.2 ) & scale_colour_gradientn(colours = rev(viridis::rocket(10)))
FeaturePlot(bc.combined.cc, features = "Satb1" , reduction = "wnn.umap", pt.size = 0.2 ) & scale_colour_gradientn(colours = rev(viridis::rocket(10)))
```

```{r}
VlnPlot(bc.combined.cc, features = "Satb1",pt.size = 0, group.by = "annotation") & geom_boxplot(width=0.1, fill="white")
VlnPlot(b6.combined.cc, features = "Satb1",pt.size = 0, group.by = "stage") & geom_boxplot(width=0.1, fill="white")
```

```{r}
Idents(bc.combined.cc) <- "OTIIstatus"
VlnPlot(bc.combined.cc, features = "Satb1",idents = c("OTII", "OTII - multichains","non OTII"), pt.size = 0, group.by = "OTIIstatus") & geom_boxplot(width=0.1, fill="white")
Idents(b6.combined.cc) <- "OTIIstatus"
VlnPlot(b6.combined.cc, features = "Satb1",idents = c("OTII", "OTII - multichains","non OTII"), pt.size = 0, group.by = "OTIIstatus") & geom_boxplot(width=0.1, fill="white")
```

```{r}
Idents(bc.combined.cc) <- "sig.annotation"
VlnPlot(bc.combined.cc, features = "Satb1", pt.size = 0,idents = c("DN","DPblast","DPsm","DPsmCD25+","DPsmCCR7+","DP69", "DPblast Myc+","DPsm Myc+","DP69 Myc+","DP to SP","SP8CD69+","SP8","SP4CD69+","SP4"), group.by = "sig.annotation") & geom_boxplot(width=0.1, fill="white")
Idents(b6.combined.cc) <- "sig.annotation"
VlnPlot(b6.combined.cc, features = "Satb1",idents = c("DN-early DP","DPsm","SP4 immature", "SP4 mature", "DPblast-Myc+","DPsm-Myc+","DP69-Myc+","SP4-Myc+"), pt.size = 0, group.by = "sig.annotation") & geom_boxplot(width=0.1, fill="white")
```

#Arpp21

```{r}
VlnPlot(bc.combined.cc, features = "Arpp21",pt.size = 0, group.by = "annotation") & geom_boxplot(width=0.1, fill="white")
VlnPlot(b6.combined.cc, features = "Arpp21",pt.size = 0, group.by = "stage") & geom_boxplot(width=0.1, fill="white")

Idents(b6.combined.cc) <-"stage"
VlnPlot(b6.combined.cc, features = "Arpp21",group.by= "sig.annotation", idents = c("Thymus-WT"))
VlnPlot(b6.combined.cc, features = "Arpp21",group.by= "sig.annotation", idents = c("Thymus-PreTum"))
VlnPlot(b6.combined.cc, features = "Arpp21",group.by= "sig.annotation", idents = c("Thymus-Tum"))
VlnPlot(b6.combined.cc, features = "Arpp21",group.by= "sig.annotation",split.by = "stage", idents = c("Thymus-WT","Thymus-PreTum","Thymus-Tum"))
```

#Adam12

```{r}
VlnPlot(b6.combined.cc, features = "Adam12",pt.size = 0, group.by = "sig.annotation") & geom_boxplot(width=0.1, fill="white")
VlnPlot(bc.combined.cc, features = "Adam12",pt.size = 0, group.by = "sig.annotation") & geom_boxplot(width=0.1, fill="white")
VlnPlot(b6.combined.cc, features = "Adam12",pt.size = 0, group.by = "stage") & geom_boxplot(width=0.1, fill="white")
```

# Bach2

```{r}
VlnPlot(b6.combined.cc, features = "Bach2",pt.size = 0, group.by = "sig.annotation") & geom_boxplot(width=0.1, fill="white")
VlnPlot(bc.combined.cc, features = "Bach2",pt.size = 0, group.by = "sig.annotation") & geom_boxplot(width=0.1, fill="white")
VlnPlot(b6.combined.cc, features = "Adam12",pt.size = 0, group.by = "stage") & geom_boxplot(width=0.1, fill="white")
```
#Mtus2
```{r}
VlnPlot(b6.combined.cc, features = "Mtus2",pt.size = 0, group.by = "sig.annotation") & geom_boxplot(width=0.1, fill="white")
VlnPlot(bc.combined.cc, features = "Mtus2",pt.size = 0, group.by = "sig.annotation") & geom_boxplot(width=0.1, fill="white")
```
#Trp53inp1
```{r}
VlnPlot(b6.combined.cc, features = "Btg2",pt.size = 0.1, group.by = "sig.annotation") & geom_boxplot(width=0.1, fill="white")
VlnPlot(bc.combined.cc, features = "Btg2",pt.size = 0.1, group.by = "sig.annotation") & geom_boxplot(width=0.1, fill="white")
```

```{r}
FeaturePlot(bc.combined.cc, features = c("Cd4","Satb1","Malat1","Ccnd3","Prkcb","Btg2","Ccr9","Itk","Grap2","Aff1","Itpr2","Runx1","Tapt1","Mtus2","Trp53inp1","Lyst"), reduction = "wnn.umap", pt.size = 0.2, ncol = 4, order = T)& scale_colour_gradientn(colours = rev(viridis::rocket(10))) &
    NoLegend() + 
    theme_void()+
    theme(plot.title = element_text(hjust = 0.5))
```

